diff -ur SuperLU_DIST_2.3/SRC/Makefile SuperLU_DIST_2.3-okk/Makefile
--- SuperLU_DIST_2.3/SRC/Makefile	2008-09-13 17:13:18.000000000 +0200
+++ SuperLU_DIST_2.3-okk/Makefile	2010-02-16 08:24:45.000000000 +0100
@@ -51,6 +51,8 @@
 ZSLUSRC	= dcomplex.o zlangs.o zgsequ.o zlaqgs.o zutil.o \
 	  zmemory.o zmyblas2.o dmemory.o zsp_blas2.o zsp_blas3.o
 
+#ZSLUSRCdemory= dmemory.o
+
 #
 # Routines for double precision parallel SuperLU
 # DPLUSRC = dldperm.o ddistribute.o pdgstrf.o pdgstrs_Bglobal.o 
@@ -69,7 +71,7 @@
 	  pzgsequ.o pzlaqgs.o pzgsrfs.o pzgsmv.o pzgstrs_lsum.o \
 	  pzsymbfact_distdata.o
 
-all:  double complex16
+all:  doublecomplex16
 
 double: $(DSLUSRC) $(DPLUSRC) $(ALLAUX) $(LAAUX)
 	$(ARCH) $(ARCHFLAGS) $(DSUPERLULIB) \
@@ -81,6 +83,10 @@
 		$(ZSLUSRC) $(ZPLUSRC) $(ALLAUX) $(LAAUX)
 	$(RANLIB) $(DSUPERLULIB)
 
+doublecomplex16: $(DSLUSRC) $(DPLUSRC) $(ZSLUSRC) $(ZPLUSRC) $(ALLAUX) $(LAAUX)
+	$(ARCH) $(ARCHFLAGS) $(DSUPERLULIB) \
+		$(DSLUSRC) $(DPLUSRC) $(ZSLUSRC) $(ZPLUSRC) $(ALLAUX) $(LAAUX)
+	$(RANLIB) $(DSUPERLULIB)
 
 ##################################
 # Do not optimize these routines #
