diff -ur SuperLU_DIST_3.0.orig/SRC/pdgstrf.c SuperLU_DIST_3.0/SRC/pdgstrf.c
--- SuperLU_DIST_3.0.orig/SRC/pdgstrf.c	2011-10-11 20:45:40.000000000 +0000
+++ SuperLU_DIST_3.0/SRC/pdgstrf.c	2013-02-13 12:51:45.691434748 +0000
@@ -2597,7 +2597,7 @@
 		    MPI_Isend(ublk_ptr, u_diag_cnt, MPI_DOUBLE, pr,
 			      tag, comm, U_diag_blk_send_req + pr);
 		}
-	    U_diag_blk_send_req[krow] = 1; /* flag outstanding Isend */
+	    U_diag_blk_send_req[krow] = (MPI_Request) 1; /* flag outstanding Isend */
 	}
 
     } else  { /* non-diagonal process */
Only in SuperLU_DIST_3.0/SRC: pdgstrf.o
diff -ur SuperLU_DIST_3.0.orig/SRC/pzgstrf.c SuperLU_DIST_3.0/SRC/pzgstrf.c
--- SuperLU_DIST_3.0.orig/SRC/pzgstrf.c	2011-10-11 20:45:40.000000000 +0000
+++ SuperLU_DIST_3.0/SRC/pzgstrf.c	2013-02-13 12:54:00.212442635 +0000
@@ -2543,7 +2543,7 @@
 		    MPI_Isend(ublk_ptr, u_diag_cnt, SuperLU_MPI_DOUBLE_COMPLEX, pr,
 			      tag, comm, U_diag_blk_send_req + pr);
 		}
-	    U_diag_blk_send_req[krow] = 1; /* flag outstanding Isend */
+	    U_diag_blk_send_req[krow] = (MPI_Request) 1; /* flag outstanding Isend */
 	}
 
     } else  { /* non-diagonal process */
