Only in SuperLU_DIST_3.0/SRC: 1
diff -ur SuperLU_DIST_3.0.orig/SRC/dutil.c SuperLU_DIST_3.0/SRC/dutil.c
--- SuperLU_DIST_3.0.orig/SRC/dutil.c	2011-09-06 00:16:34.000000000 +0000
+++ SuperLU_DIST_3.0/SRC/dutil.c	2013-02-13 17:40:36.696123287 +0000
@@ -185,6 +185,8 @@
     if ( (dp = (double *) Astore->nzval) != NULL )
         PrintDouble5("nzval", nnz_loc, dp);
     printf("==== end CompRowLoc matrix\n");
+
+    return 0;
 }
 
 int file_dPrint_CompRowLoc_Matrix_dist(FILE *fp, SuperMatrix *A)
@@ -205,6 +207,8 @@
     if ( (dp = (double *) Astore->nzval) != NULL )
         file_PrintDouble5(fp, "nzval", nnz_loc, dp);
     fprintf(fp, "==== end CompRowLoc matrix\n");
+
+    return 0;
 }
 
 void
@@ -355,6 +359,8 @@
 	fprintf(fp, "%14e", x[i]);
     }
     fprintf(fp, "\n");
+
+    return 0;
 }
 
 /*! \brief Print the blocks in the factored matrix L.
@@ -461,6 +467,8 @@
   file_PrintInt10(fp, "ptr_ind_torecv", procs+1, gsmv_comm->ptr_ind_torecv);
   file_PrintInt10(fp, "SendCounts", procs, gsmv_comm->SendCounts);
   file_PrintInt10(fp, "RecvCounts", procs, gsmv_comm->RecvCounts);
+
+  return 0;
 }
 
 
Binary files SuperLU_DIST_3.0.orig/SRC/dutil.o and SuperLU_DIST_3.0/SRC/dutil.o differ
diff -ur SuperLU_DIST_3.0.orig/SRC/pddistribute.c SuperLU_DIST_3.0/SRC/pddistribute.c
--- SuperLU_DIST_3.0.orig/SRC/pddistribute.c	2011-09-06 00:16:34.000000000 +0000
+++ SuperLU_DIST_3.0/SRC/pddistribute.c	2013-02-13 17:44:50.198138179 +0000
@@ -304,6 +304,8 @@
     CHECK_MALLOC(iam, "Exit dReDistribute_A()");
 #endif
  
+    return 0;
+
 } /* dReDistribute_A */
 
 float
Binary files SuperLU_DIST_3.0.orig/SRC/pddistribute.o and SuperLU_DIST_3.0/SRC/pddistribute.o differ
diff -ur SuperLU_DIST_3.0.orig/SRC/psymbfact.c SuperLU_DIST_3.0/SRC/psymbfact.c
--- SuperLU_DIST_3.0.orig/SRC/psymbfact.c	2011-09-06 00:16:34.000000000 +0000
+++ SuperLU_DIST_3.0/SRC/psymbfact.c	2013-02-13 17:47:47.931004764 +0000
@@ -2267,6 +2267,8 @@
   
   for (i = fstVtx_toUpd; i < nvtcs_toUpd; i++)
     marker[i] = 0;
+
+    return 0;
 }
 
 static int_t
@@ -4005,6 +4007,8 @@
   if (newelts_U) SUPERLU_FREE (newelts_U);
   if (PS->szDnsSep < mem_dnsCS)
     PS->szDnsSep = mem_dnsCS;
+
+  return 0;
 }
 
 /*! \brief
@@ -4534,6 +4538,8 @@
   if (request_snd != NULL) SUPERLU_FREE (request_snd);
   if (request_rcv != NULL) SUPERLU_FREE (request_rcv);
   if (status != NULL) SUPERLU_FREE (status);
+
+  return 0;
 }
 
 static void
Binary files SuperLU_DIST_3.0.orig/SRC/psymbfact.o and SuperLU_DIST_3.0/SRC/psymbfact.o differ
diff -ur SuperLU_DIST_3.0.orig/SRC/pzdistribute.c SuperLU_DIST_3.0/SRC/pzdistribute.c
--- SuperLU_DIST_3.0.orig/SRC/pzdistribute.c	2011-09-06 00:16:34.000000000 +0000
+++ SuperLU_DIST_3.0/SRC/pzdistribute.c	2013-02-13 17:53:06.269023492 +0000
@@ -303,6 +303,7 @@
     CHECK_MALLOC(iam, "Exit zReDistribute_A()");
 #endif
  
+    return 0;
 } /* zReDistribute_A */
 
 float
Binary files SuperLU_DIST_3.0.orig/SRC/pzdistribute.o and SuperLU_DIST_3.0/SRC/pzdistribute.o differ
diff -ur SuperLU_DIST_3.0.orig/SRC/util.c SuperLU_DIST_3.0/SRC/util.c
--- SuperLU_DIST_3.0.orig/SRC/util.c	2011-10-11 21:14:16.000000000 +0000
+++ SuperLU_DIST_3.0/SRC/util.c	2013-02-13 17:54:31.042028470 +0000
@@ -490,6 +490,7 @@
     gstrs_comm->ptr_to_ibuf = ptr_to_ibuf;
     gstrs_comm->ptr_to_dbuf = ptr_to_ibuf + procs;
 
+    return 0;
 } /* PXGSTRS_INIT */
 
 
@@ -763,6 +764,8 @@
 	fprintf(fp, "%6d", x[i]);
     }
     fprintf(fp, "\n");
+
+    return 0;
 }
 
 int_t
Binary files SuperLU_DIST_3.0.orig/SRC/util.o and SuperLU_DIST_3.0/SRC/util.o differ
diff -ur SuperLU_DIST_3.0.orig/SRC/zutil.c SuperLU_DIST_3.0/SRC/zutil.c
--- SuperLU_DIST_3.0.orig/SRC/zutil.c	2011-09-06 00:16:34.000000000 +0000
+++ SuperLU_DIST_3.0/SRC/zutil.c	2013-02-13 17:56:24.618035143 +0000
@@ -184,6 +184,8 @@
     if ( (dp = (doublecomplex *) Astore->nzval) != NULL )
         PrintDoublecomplex("nzval", nnz_loc, dp);
     printf("==== end CompRowLoc matrix\n");
+
+    return 0;
 }
 
 int file_zPrint_CompRowLoc_Matrix_dist(FILE *fp, SuperMatrix *A)
@@ -204,6 +206,8 @@
     if ( (dp = (doublecomplex *) Astore->nzval) != NULL )
         file_PrintDoublecomplex(fp, "nzval", nnz_loc, dp);
     fprintf(fp, "==== end CompRowLoc matrix\n");
+
+    return 0;
 }
 
 void
@@ -351,6 +355,8 @@
     fprintf(fp, "%10s:\tReal\tImag\n", name);
     for (i = 0; i < len; ++i)
 	fprintf(fp, "\t%d\t%.4f\t%.4f\n", i, x[i].r, x[i].i);
+
+    return 0;
 }
 
 /*! \brief Print the blocks in the factored matrix L.
@@ -457,6 +463,8 @@
   file_PrintInt10(fp, "ptr_ind_torecv", procs+1, gsmv_comm->ptr_ind_torecv);
   file_PrintInt10(fp, "SendCounts", procs, gsmv_comm->SendCounts);
   file_PrintInt10(fp, "RecvCounts", procs, gsmv_comm->RecvCounts);
+
+  return 0;
 }
 
 
Binary files SuperLU_DIST_3.0.orig/SRC/zutil.o and SuperLU_DIST_3.0/SRC/zutil.o differ
