diff -u mshmet.2009.09.16/sources/mshmet.c mshmet.2009.09.16-okk/mshmet.c
--- mshmet.2009.09.16/sources/mshmet.c	2009-08-28 14:56:26.000000000 +0200
+++ mshmet.2009.09.16-okk/mshmet.c	2010-03-22 08:51:09.000000000 +0100
@@ -8,7 +8,7 @@
 #include "mshmet.h"
 #include "compil.date"
 
-char     idir[5]     = {0,1,2,0,1};
+//char     idir[5]     = {0,1,2,0,1};
 mytime   ctim[TIMEMAX];
 
 
@@ -78,7 +78,7 @@
   info->width  = 0.05;
   info->ddebug = 0;
   info->imprim = -99;
-	info->nlis   = 0;
+  info->nlis   = 0;
 
   i = 1;
   while ( i < argc ) {

diff -u mshmet.2009.09.16/sources/mshmet.h mshmet.2009.09.16-okk/mshmet.h
--- mshmet.2009.09.16/sources/mshmet.h	2009-08-27 17:26:33.000000000 +0200
+++ mshmet.2009.09.16-okk/mshmet.h	2010-03-24 18:57:02.000000000 +0100
@@ -13,6 +13,8 @@
 #include "chrono.h"
 #include "memory.h"
 #include "libmesh5.h"
+#include "mshmetlib.h"
+#include "mshmetlib-internal.h"
 #include "eigenv.h"
 
 #define MS_VER   "2.3c"
@@ -33,8 +35,8 @@
 #define MS_MAX(a,b)   ( ((a) < (b)) ? (b) : (a) )
 #define MS_MIN(a,b)   ( ((a) < (b)) ? (a) : (b) )
 
-char idir[5];
-
+const static char idir[5] = {0,1,2,0,1};
+/*
 typedef struct {
   double         aire,rins;
   double         c[3];
@@ -88,7 +90,7 @@
   double         hes[6];
 } Deriv;
 typedef Deriv * pDeriv;
-
+*/
 
 /* prototypes */
 int eigen_3d(int ,double *,double *,double v[3][3]);

diff -u mshmet.2009.09.16/sourceslib/hash.c mshmet.2009.09.16-okk/hash.c
--- mshmet.2009.09.16/sourceslib/hash.c	2010-03-25 17:02:24.000000000 +0100
+++ mshmet.2009.09.16-okk/hash.c	2010-03-25 17:04:09.000000000 +0100
@@ -24,7 +24,7 @@
   hsize = mesh->ne;
 
   /* init */
-  inival = 2 << 30;
+  inival = 2 << 29; // initval = 2 << 30; problem with gcc-4.4 -O3 for this initialization of inival for linking with freefem++
   for (k=0; k<=mesh->ne; k++)
     hcode[k] = -inival;
 
@@ -52,7 +52,7 @@
       hcode[key] = -iadr;
     }
   }
-
+  
   /* set adjacency */
   for (l=4*mesh->ne; l>0; l--) {
     if ( link[l] >= 0 )  continue;
@@ -68,7 +68,7 @@
     mins = MS_MIN(mins,pt->v[i3]);
     maxs = MS_MAX(pt->v[i1],pt->v[i2]);
     maxs = MS_MAX(maxs,pt->v[i3]);
-
+    
     /* accross link */
     ll = -link[l];
     pp = 0;
@@ -142,7 +142,7 @@
   hvoy  = (unsigned char*)hcode;
 
   /* init */
-  inival = 2 << 30;
+  inival = 2 << 29; // initval = 2 << 30; problem with gcc-4.4 -O3 for this initialization of inival for linking with freefem++
   for (k=0; k<=mesh->nt; k++)
     hcode[k] = -inival;
 

 
 
