diff -u yams2.2010.02.22/sourceslib/blinde.c yams2.2010.02.22-okk/blinde.c
--- yams2.2010.02.22/sourceslib/blinde.c	2010-03-22 10:54:43.000000000 +0100
+++ yams2.2010.02.22-okk/blinde.c	2010-03-22 10:27:27.000000000 +0100
@@ -36,7 +36,7 @@
   ida =                          id[0]*10 + id[1];
   /*printf("date = %d %d %d\n",iy,im,ida); */
   if ( YMD(iy,im,ida) > YMD(MAX_YEAR,MAX_MONTH,MAX_DAY) ) {
-    fprintf(stdout,"  -- YAMS (LJLL), Version %s (%s)\n",VERSION,RELEASE);
+    fprintf(stdout,"  -- YAMS (LJLL), Version %s (%s)\n",YAMS_VERSION,RELEASE);
     fprintf(stdout,"     Copyright (C) LJLL, 1999-2006.\n\n");
     fprintf(stdout," ## Expiration date reached. Sorry.\n");
     fprintf(stdout,"    Please contact the author.\n");

diff -u yams2.2010.02.22/sourceslib/defines.h yams2.2010.02.22-okk/defines.h
--- yams2.2010.02.22/sourceslib/defines.h	2010-03-22 11:20:01.000000000 +0100
+++ yams2.2010.02.22-okk/defines.h	2010-03-22 11:21:42.000000000 +0100
@@ -18,7 +18,7 @@
 /* current version */
 #define DEFAULT_FILE    "DEFAULT.yams"
 #define RELEASE         "oct, 2006"
-#define VERSION         "2.4 b"
+#define YAMS_VERSION    "2.4 b"
 #define COPYRIGHT       "Copyright (C) LJLL, 1999-2006"
 
 #define REL    1


diff -u yams2.2010.02.22/sourceslib/extern.h yams2.2010.02.22-okk/extern.h
--- yams2.2010.02.22/sourceslib/extern.h	2010-03-22 10:54:44.000000000 +0100
+++ yams2.2010.02.22-okk/extern.h	2010-03-22 10:27:58.000000000 +0100
@@ -4,6 +4,7 @@
 
 
 #ifndef __YAMS
+#ifndef __YAMSLIB
   extern Error      yerr;
   extern Info       info;
   extern Options    opts;
@@ -14,7 +15,7 @@
   extern short     imprim;
   extern ubyte     ddebug;
 #endif
-
+#endif
 
 #ifdef __cplusplus
 }

diff -u yams2.2010.02.22/sourceslib/inout.c yams2.2010.02.22-okk/inout.c
--- yams2.2010.02.22/sourceslib/inout.c	2010-03-22 10:54:43.000000000 +0100
+++ yams2.2010.02.22-okk/inout.c	2010-03-22 10:29:00.000000000 +0100
@@ -577,7 +577,7 @@
     ppt->tag |= M_UNUSED;
     ppt->flag = ppt->color = 0;
   }
-
+  printf("sm->connex %d\n",sm->connex);
   if ( sm->connex > 0 ) {
     for (k=1; k<=sm->ne; k++) {
       pt1 = &sm->tria[k];
@@ -674,8 +674,8 @@
         natv++;
         if ( ppt->tag & M_CORNER )  tatv++;
       }
-      if ( !gs->new )  gs->new = ++nn;
-      if ( !gt->new )  gt->new = ++nt;
+      if ( !gs->newnum )  gs->newnum = ++nn;
+      if ( !gt->newnum )  gt->newnum = ++nt;
       if ( !pt1->edg[i] && pt1->tag[i] == M_NOTAG )  continue;
       else if ( pt1->adj[i] && (k > pt1->adj[i]) )   continue;
       nedge++;
@@ -860,12 +860,12 @@
   nn = nbl = 0;
   for (k=1; k<=sm->nvmax; k++) {
     gs = &sm->geom[k];
-    if ( gs->new > 0 ) {
+    if ( gs->newnum > 0 ) {
       iadr = nbl * 3;
       tabf[iadr+0] = gs->vn[0];
       tabf[iadr+1] = gs->vn[1];
       tabf[iadr+2] = gs->vn[2];
-      gs->new = ++nn;
+      gs->newnum = ++nn;
       ++nbl;
       if ( nbl == NMAX ) {
         LM_write_field(&ms, LM_Normals, nbl, tabf);
@@ -886,7 +886,7 @@
         iadr = nbl * 2;
         gs = &sm->geom[pt1->vn[i]];
         tabi[iadr+0] = ppt->tmp;
-        tabi[iadr+1] = gs->new;
+        tabi[iadr+1] = gs->newnum;
         ppt->flag = 1;
         ++nbl;
         if ( nbl == NMAX ) {
@@ -912,7 +912,7 @@
         iadr = nbl * 3;
         tabi[iadr+0] = nn;
         tabi[iadr+1] = i+1;
-        tabi[iadr+2] = gs->new;
+        tabi[iadr+2] = gs->newnum;
         ++nbl;
         if ( nbl == NMAX ) {
           LM_write_field(&ms, LM_NormalAtTriangleVertices, nbl, tabi);
@@ -927,12 +927,12 @@
   nt = nbl = 0;
   for (k=1; k<=sm->ntmax; k++) {
     gt = &sm->tgte[k];
-    if ( gt->new > 0 ) {
+    if ( gt->newnum > 0 ) {
       iadr = nbl * 3;
       tabf[iadr+0] = gt->t[0];
       tabf[iadr+1] = gt->t[1];
       tabf[iadr+2] = gt->t[2];
-      gt->new  = ++nt;
+      gt->newnum  = ++nt;
       ++nbl;
       if ( nbl == NMAX ) {
         LM_write_field(&ms, LM_Tangents, nbl, tabf);
@@ -952,7 +952,7 @@
       iadr = nbl * 2;
       gt = &sm->tgte[ppt->tge];
       tabi[iadr+0]  = ppt->tmp;
-      tabi[iadr+1]  = gt->new;
+      tabi[iadr+1]  = gt->newnum;
       ppt->flag = 1;
       ++nbl;
       if ( nbl == NMAX ) {
@@ -1101,8 +1101,8 @@
         natv++;
         if ( ppt->tag & M_CORNER )  tatv++;
       }
-      if ( !gs->new )  gs->new = ++nn;
-      if ( !gt->new )  gt->new = ++nq;
+      if ( !gs->newnum )  gs->newnum = ++nn;
+      if ( !gt->newnum )  gt->newnum = ++nq;
       if ( !pq1->edg[i] && pq1->tag[i] == M_NOTAG )  continue;
       else if ( pq1->adj[i] && (k > pq1->adj[i]) )   continue;
       nedge++;
@@ -1259,12 +1259,12 @@
   nn = nbl = 0;
   for (k=1; k<=sm->nvmax; k++) {
     gs = &sm->geom[k];
-    if ( gs->new > 0 ) {
+    if ( gs->newnum > 0 ) {
       iadr = nbl * 3;
       tabf[iadr+0] = gs->vn[0];
       tabf[iadr+1] = gs->vn[1];
       tabf[iadr+2] = gs->vn[2];
-      gs->new = ++nn;
+      gs->newnum = ++nn;
       ++nbl;
       if ( nbl == NMAX ) {
         LM_write_field(&ms, LM_Normals, nbl, tabf);
@@ -1285,7 +1285,7 @@
         iadr = nbl * 2;
         gs = &sm->geom[pq1->vn[i]];
         tabi[iadr+0] = ppt->tmp;
-        tabi[iadr+1] = gs->new;
+        tabi[iadr+1] = gs->newnum;
         ppt->flag = 1;
         ++nbl;
         if ( nbl == NMAX ) {
@@ -1311,7 +1311,7 @@
     iadr = nbl * 3;
     tabi[iadr+0] = nn;
     tabi[iadr+1] = i+1;
-    tabi[iadr+2] = gs->new;
+    tabi[iadr+2] = gs->newnum;
     ++nbl;
         if ( nbl == NMAX ) {
           LM_write_field(&ms, LM_NormalAtQuadrilateralVertices, nbl, tabi);
@@ -1326,12 +1326,12 @@
   nq = nbl = 0;
   for (k=1; k<=sm->ntmax; k++) {
     gt = &sm->tgte[k];
-    if ( gt->new > 0 ) {
+    if ( gt->newnum > 0 ) {
       iadr = nbl * 3;
       tabf[iadr+0] = gt->t[0];
       tabf[iadr+1] = gt->t[1];
       tabf[iadr+2] = gt->t[2];
-      gt->new  = ++nq;
+      gt->newnum  = ++nq;
       ++nbl;
       if ( nbl == NMAX ) {
         LM_write_field(&ms, LM_Tangents, nbl, tabf);
@@ -1351,7 +1351,7 @@
       iadr = nbl * 2;
       gt = &sm->tgte[ppt->tge];
       tabi[iadr+0]  = ppt->tmp;
-      tabi[iadr+1]  = gt->new;
+      tabi[iadr+1]  = gt->newnum;
       ppt->flag = 1;
       ++nbl;
       if ( nbl == NMAX ) {
@@ -1520,8 +1520,8 @@
   /* adjust sizes */
   if ( opts.hmin < 0.0 )
     opts.hmin = max(opts.hmin,hmin);
-    if ( opts.hmax < 0.0 )
-      opts.hmax = max(opts.hmax,hmax);
+  if ( opts.hmax < 0.0 )
+    opts.hmax = max(opts.hmax,hmax);
 
   E_pop();
   return(1);

diff -u yams2.2010.02.22/sourceslib/parsop.c yams2.2010.02.22-okk/parsop.c
--- yams2.2010.02.22/sourceslib/parsop.c	2010-03-22 10:54:43.000000000 +0100
+++ yams2.2010.02.22-okk/parsop.c	2010-03-22 10:32:06.000000000 +0100
@@ -16,6 +16,6 @@
   }
 
   /* create standard parameter file */
-  fprintf(out,"#  Generated by YAMS %s\n",VERSION);
+  fprintf(out,"#  Generated by YAMS %s\n",YAMS_VERSION);
   fprintf(out,"#  Uncomment next if absolute unit desired\n");
   fprintf(out,"#Absolute\n");


diff -u yams2.2010.02.22/sourceslib/primsg.c yams2.2010.02.22-okk/primsg.c
--- yams2.2010.02.22/sourceslib/primsg.c	2010-03-22 11:22:55.000000000 +0100
+++ yams2.2010.02.22-okk/primsg.c	2010-03-22 11:24:16.000000000 +0100
@@ -18,10 +18,10 @@
     case 0000:
       if ( opts.ctrl & ISO )
         fprintf(stdout,"\n  %s\n   MODULE YAMS-LJLL : %s\n  %s\n",
-	        STR,VERSION,STR);
+	        STR,YAMS_VERSION,STR);
       else
         fprintf(stdout,"\n  %s\n   MODULE YAMS-LJLL (ANISO) : %s\n  %s\n",
-	        STR,VERSION,STR);
+	        STR,YAMS_VERSION,STR);
       return;
     case 0001: 
       fprintf(stdout,"\n  %s\n   END OF MODULE YAMS\n  %s\n\n",STR,STR);


diff -u yams2.2010.02.22/sourceslib/surf.h yams2.2010.02.22-okk/surf.h
--- yams2.2010.02.22/sourceslib/surf.h	2010-03-22 10:54:44.000000000 +0100
+++ yams2.2010.02.22-okk/surf.h	2010-03-22 10:33:04.000000000 +0100
@@ -64,13 +64,13 @@
 typedef struct geomsupp {
   float    vn[3];           /* array of vertex normals  */
   float    gap;             /* local gap value          */
-  int      new;             /* pointer to new number    */
+  int      newnum;             /* pointer to new number    */
 } GeomSupp;
 typedef GeomSupp  * pGeomSupp;
 
 typedef struct geomtge {
   float    t[3];
-  int      new;
+  int      newnum;
 } Geomtge;
 typedef Geomtge   * pGeomtge;
 
diff -u yams2.2010.02.22/sourceslib/yams.h yams2.2010.02.22-okk/yams.h
--- yams2.2010.02.22/sourceslib/yams.h	2010-03-22 10:54:44.000000000 +0100
+++ yams2.2010.02.22-okk/yams.h	2010-03-22 10:34:07.000000000 +0100
@@ -6,12 +6,17 @@
 #include <string.h>
 
 #include "chrono.h"
-#include "edge.h"
-#include "error.h"
-#include "hash.h"
-#include "stack.h"
-#include "info.h"
 #include "memory.h"
-#include "option.h"
-#include "surf.h"
+
+//#include "edge.h"
+//#include "error.h"
+//#include "hash.h"
+//#include "stack.h"
+//#include "info.h"
+//#include "memory.h"
+//#include "option.h"
+//#include "surf.h"
+
+#include "yamslib.h"
+#include "yamslib_internal.h"
 #include "global.h"

diff -u yams2.2010.02.22/sourceslib/yams0.c yams2.2010.02.22-okk/yams0.c
--- yams2.2010.02.22/sourceslib/yams0.c	2010-03-22 10:54:43.000000000 +0100
+++ yams2.2010.02.22-okk/yams0.c	2010-03-22 10:34:28.000000000 +0100
@@ -24,7 +24,7 @@
   if ( imprim ) {
     if ( ecp )
       fprintf(stdout,"  ## Version:  Ecole Centrale de Paris ##\n\n");
-    fprintf(stdout,"  -- YAMS (LJLL) Version %s (%s)\n",VERSION,RELEASE);
+    fprintf(stdout,"  -- YAMS (LJLL) Version %s (%s)\n",YAMS_VERSION,RELEASE);
     fprintf(stdout,"     %s.\n",COPYRIGHT);
     fprintf(stdout,"     compiled: %s.\n\n",COMPIL);
   }

diff -u yams2.2010.02.22/sourceslib/zaldy1.c yams2.2010.02.22-okk/zaldy1.c
--- yams2.2010.02.22/sourceslib/zaldy1.c	2010-03-22 10:54:43.000000000 +0100
+++ yams2.2010.02.22-okk/zaldy1.c	2010-03-22 10:34:50.000000000 +0100
@@ -45,13 +45,20 @@
     }
   }
   else {
+    int bytes = sizeof(Point) \
+              + sizeof(Metric) \
+              + 2   * sizeof(Triangle) \
+              + 1.5 * sizeof(GeomSupp) \
+              + 0.2 * sizeof(Geomtge) \
+              + 3.7 * sizeof(Hashtable);
+    /*
     int bytes = sizeof(struct spoint) \
               + sizeof(struct metric) \
               + 2   * sizeof(struct striangle) \
               + 1.5 * sizeof(struct geomsupp) \
               + 0.2 * sizeof(struct geomtge) \
               + 3.7 * sizeof(struct shashtab);
-
+    */
     sm->npmax = (int)((double)memory / bytes * million);
     sm->npmax = max(sm->npmax,npmax);
     sm->nemax = max(2 * sm->npmax,nemax);

diff -u yams2.2010.02.22/sourceslib/zaldy3.c yams2.2010.02.22-okk/zaldy3.c
--- yams2.2010.02.22/sourceslib/zaldy3.c	2010-03-22 10:54:43.000000000 +0100
+++ yams2.2010.02.22-okk/zaldy3.c	2010-03-22 10:35:06.000000000 +0100
@@ -34,7 +34,7 @@
     for (j=sm->nvmax; j<nvsize; j++) {
       sm->geom[j].vn[0] = sm->geom[j].vn[1] = sm->geom[j].vn[2] = 0.;
       sm->geom[j].gap   = 1.;
-      sm->geom[j].new   = 0;
+      sm->geom[j].newnum   = 0;
     }
     sm->nvmax = nvsize;
     yerr.inderr[0] = sm->nvmax;
