--- metis-4.0/Programs/kmetis.c.orig	2013-01-26 21:01:39.898052768 +0000
+++ metis-4.0/Programs/kmetis.c	2013-01-26 21:01:52.186053491 +0000
@@ -19,7 +19,7 @@
 /*************************************************************************
 * Let the game begin
 **************************************************************************/
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
   int i, nparts, options[10];
   idxtype *part;
@@ -100,6 +100,8 @@
 
 
   GKfree(&graph.xadj, &graph.adjncy, &graph.vwgt, &graph.adjwgt, &part, LTERM);
+
+  return 0;
 }  
 
 
--- metis-4.0/Programs/pmetis.c.orig	2013-01-26 22:42:46.375724925 +0000
+++ metis-4.0/Programs/pmetis.c	2013-01-26 22:43:33.744727711 +0000
@@ -19,7 +19,7 @@
 /*************************************************************************
 * Let the game begin
 **************************************************************************/
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
   int i, nparts, options[10];
   idxtype *part;
@@ -100,6 +100,8 @@
 
 
   GKfree(&graph.xadj, &graph.adjncy, &graph.vwgt, &graph.adjwgt, &part, LTERM);
+  
+  return 0;
 }  
 
 
--- ./metis-4.0/Programs/graphchk.c.orig	2013-01-26 23:00:11.167253583 +0000
+++ ./metis-4.0/Programs/graphchk.c	2013-01-26 23:00:38.003255160 +0000
@@ -19,7 +19,7 @@
 /*************************************************************************
 * Let the game begin
 **************************************************************************/
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
   GraphType graph;
   char filename[256];
@@ -53,6 +53,8 @@
 
 
   GKfree(&graph.xadj, &graph.adjncy, &graph.vwgt, &graph.adjwgt, LTERM);
+  
+  return 0;
 }  
 
 
--- ./metis-4.0/Programs/mesh2dual.c.orig	2013-01-26 23:02:14.677260844 +0000
+++ ./metis-4.0/Programs/mesh2dual.c	2013-01-26 23:02:45.081262631 +0000
@@ -20,7 +20,7 @@
 /*************************************************************************
 * Let the game begin
 **************************************************************************/
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
   int i, j, ne, nn, etype, numflag=0;
   idxtype *elmnts, *xadj, *adjncy;
@@ -67,6 +67,7 @@
 
   GKfree(&elmnts, &xadj, &adjncy, LTERM);
 
+  return 0;
 }
 
 
--- ./metis-4.0/Programs/mesh2nodal.c.orig	2013-01-26 23:03:35.671265601 +0000
+++ ./metis-4.0/Programs/mesh2nodal.c	2013-01-26 23:03:59.234266985 +0000
@@ -20,7 +20,7 @@
 /*************************************************************************
 * Let the game begin
 **************************************************************************/
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
   int i, j, ne, nn, etype, numflag=0;
   idxtype *elmnts, *xadj, *adjncy;
@@ -67,6 +67,7 @@
 
   GKfree(&elmnts, &xadj, &adjncy, LTERM);
 
+  return 0;
 }
 
 
--- ./metis-4.0/Programs/oemetis.c.orig	2013-01-26 23:04:51.689270079 +0000
+++ ./metis-4.0/Programs/oemetis.c	2013-01-26 23:05:15.686271479 +0000
@@ -19,7 +19,7 @@
 /*************************************************************************
 * Let the game begin
 **************************************************************************/
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
   int i, options[10];
   idxtype *perm, *iperm;
@@ -90,6 +90,8 @@
 
 
   GKfree(&graph.xadj, &graph.adjncy, &perm, &iperm, LTERM);
+  
+  return 0;
 }  
 
 
--- ./metis-4.0/Programs/onmetis.c.orig	2013-01-26 23:06:04.908274368 +0000
+++ ./metis-4.0/Programs/onmetis.c	2013-01-26 23:06:26.011275618 +0000
@@ -19,7 +19,7 @@
 /*************************************************************************
 * Let the game begin
 **************************************************************************/
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
   int i, options[10];
   idxtype *perm, *iperm;
@@ -90,6 +90,8 @@
 
 
   GKfree(&graph.xadj, &graph.adjncy, &perm, &iperm, LTERM);
+
+  return 0;
 }  
 
 
--- ./metis-4.0/Programs/partdmesh.c.orig	2013-01-26 23:07:09.315278152 +0000
+++ ./metis-4.0/Programs/partdmesh.c	2013-01-26 23:07:33.761279588 +0000
@@ -20,7 +20,7 @@
 /*************************************************************************
 * Let the game begin
 **************************************************************************/
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
   int i, j, ne, nn, etype, numflag=0, nparts, edgecut;
   idxtype *elmnts, *epart, *npart;
@@ -88,6 +88,7 @@
 
   GKfree(&elmnts, &epart, &npart, LTERM);
 
+  return 0;
 }
 
 
--- ./metis-4.0/Programs/partnmesh.c.orig	2013-01-26 23:09:04.858284939 +0000
+++ ./metis-4.0/Programs/partnmesh.c	2013-01-26 23:09:45.701287342 +0000
@@ -20,7 +20,7 @@
 /*************************************************************************
 * Let the game begin
 **************************************************************************/
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
   int i, j, ne, nn, etype, numflag=0, nparts, edgecut;
   idxtype *elmnts, *epart, *npart;
@@ -88,6 +88,7 @@
 
   GKfree(&elmnts, &epart, &npart, LTERM);
 
+  return 0;
 }
 
 
