--- ParMetis-3.1.1/ParMETISLib/stdheaders.h	2008-09-16 23:39:19.000000000 +0200
+++ ParMetis-3.1.1/ParMETISLib/stdheaders.h-ok	2010-03-30 11:51:17.000000000 +0200
@@ -11,10 +11,16 @@
  * $Id$
  */
 
-
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#ifdef HAVE_MALLOC_H
+#include <malloc.h>
+#endif
+#else
+#include <malloc.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
-#include <malloc.h>
 #include <string.h>
 #include <ctype.h>
 #include <math.h>
