--- a/parmetis-4.0.3/metis/GKlib/gk_arch.h	2015-03-12 18:55:30.000000000 +0100
+++ b/parmetis-4.0.3/metis/GKlib/gk_arch.h	2015-03-12 20:27:25.000000000 +0100
@@ -41,7 +41,9 @@
 #endif
   #include <inttypes.h>
   #include <sys/types.h>
-  #include <sys/resource.h>
+#ifndef __WIN32__
+  #include <sys/resource.h>
+#endif 
   #include <sys/time.h>
 #endif
 
--- a/parmetis-4.0.3/metis/GKlib/gk_getopt.h	2013-03-30 17:24:50.000000000 +0100
+++ b/parmetis-4.0.3/metis/GKlib/gk_getopt.h	2015-03-26 14:26:48.000000000 +0100
@@ -52,11 +52,11 @@
 
 
 /* Function prototypes */
-extern int gk_getopt(int __argc, char **__argv, char *__shortopts);
-extern int gk_getopt_long(int __argc, char **__argv, char *__shortopts,
-              struct gk_option *__longopts, int *__longind);
-extern int gk_getopt_long_only (int __argc, char **__argv,
-              char *__shortopts, struct gk_option *__longopts, int *__longind);
+extern int gk_getopt(int argc, char **argv, char *shortopts);
+extern int gk_getopt_long(int argc, char **argv, char *shortopts,
+              struct gk_option *longopts, int *longind);
+extern int gk_getopt_long_only (int argc, char **argv,
+              char *shortopts, struct gk_option *longopts, int *longind);
 
 
 
