diff -ur scotch_5.1_esmumps/src/esmumps/Makefile scotch-okk/Makefile 
--- scotch_5.1_esmumps/src/esmumps/Makefile	2009-04-27 11:22:08.000000000 +0200
+++ scotch-okk/Makefile	2012-05-09 10:00:46.765713788 +0200
@@ -44,7 +44,7 @@
 				$(CC) $(CFLAGS) -I$(includedir) -c $(<) -o $(@)
 
 %$(EXE)	:	%.c
-		 		$(CC) $(CFLAGS) -I$(includedir) $(<) -o $(@) -L$(libdir) $(LDFLAGS) -L. -l$(ESMUMPSLIB) -l$(SCOTCHLIB) -l$(SCOTCHLIB)errexit
+		 		$(CC) $(CFLAGS) -I$(includedir) $(<) -o $(@) -L$(libdir) -L. -l$(ESMUMPSLIB) -l$(SCOTCHLIB) -l$(SCOTCHLIB)errexit $(LDFLAGS) 
 
 ##
 ##  Project rules.

