--- ./scotch_6.0.4/src/esmumps/Makefile~	2015-03-14 00:32:06.000000000 +0100
+++ ./scotch_6.0.4/src/esmumps/Makefile	2015-09-24 16:55:41.000000000 +0200
@@ -44,7 +44,7 @@
 				$(CC) $(CFLAGS) $(CLIBFLAGS) -I$(includedir) -c $(<) -o $(@)
 
 %$(EXE)	:	%.c
-		 		$(CC) $(CFLAGS) -I$(includedir) $(<) -o $(@) -L$(libdir) $(LDFLAGS) -L. -l$(ESMUMPSLIB) -l$(SCOTCHLIB) -lscotch -l$(SCOTCHLIB)errexit -lm
+		 		$(CC) $(CFLAGS) -I$(includedir) $(<) -o $(@) -L$(libdir) $(LDFLAGS) -L. -l$(ESMUMPSLIB) -l$(SCOTCHLIB) -lscotch -l$(SCOTCHLIB)errexit -lm $(LDLIBS)
 
 ##
 ##  Project rules.
