--- a/scotch_6.0.3/src/esmumps/Makefile	2012-12-01 15:29:29.000000000 +0100
+++ b/scotch_6.0.3/src/esmumps/Makefile	2015-03-26 22:35:28.000000000 +0100
@@ -58,7 +58,7 @@
 					main_esmumps$(EXE)
 
 ptscotch			:	clean
-					$(MAKE) CFLAGS="$(CFLAGS) -DSCOTCH_PTSCOTCH" CC=$(CCP) SCOTCHLIB=ptscotch ESMUMPSLIB=ptesmumps	\
+					$(MAKE) CFLAGS="$(CFLAGS) -DSCOTCH_PTSCOTCH" CC="$(CCP)" SCOTCHLIB=ptscotch ESMUMPSLIB=ptesmumps	\
 					libesmumps$(LIB)										\
 					main_esmumps$(EXE)
 
