diff -rpu MUMPS_4.9.2-old/src/mumps_orderings.c MUMPS_4.9.2-new/src/mumps_orderings.c
--- MUMPS_4.9.2-old/src/mumps_orderings.c	2009-11-05 08:05:21.000000000 +0100
+++ MUMPS_4.9.2-new/src/mumps_orderings.c	2010-04-22 15:34:25.000000000 +0200
@@ -343,7 +343,7 @@ MUMPS_SCOTCH( const int * const  n,
 #include "mpi.h"
 #include <stdio.h>
 #include "ptscotch.h"
-int mumps_dgraphinit( SCOTCH_Dgraph *, MPI_Fint *, MPI_Fint *);
+void mumps_dgraphinit( SCOTCH_Dgraph *, MPI_Fint *, MPI_Fint *);
 #define MUMPS_DGRAPHINIT \
   F_SYMBOL(dgraphinit,DGRAPHINIT)
 void MUMPS_CALL
