--- SuperLU_DIST_3.0/SRC/util.c.orig	2013-02-20 10:23:45.674505816 +0000
+++ SuperLU_DIST_3.0/SRC/util.c	2013-02-20 10:24:09.001507187 +0000
@@ -662,8 +662,8 @@
 
     do {
 	++(*num_diag_procs);
-	i = (++i) % nprow;
-	j = (++j) % npcol;
+	i = (i + 1) % nprow;
+	j = (j + 1) % npcol;
 	pkk = PNUM( i, j, grid );
     } while ( pkk != 0 ); /* Until wrap back to process 0 */
     if ( !(*diag_procs = intMalloc_dist(*num_diag_procs)) )
@@ -673,8 +673,8 @@
     for (i = j = k = 0; k < *num_diag_procs; ++k) {
 	pkk = PNUM( i, j, grid );
 	(*diag_procs)[k] = pkk;
-	i = (++i) % nprow;
-	j = (++j) % npcol;
+	i = (i + 1) % nprow;
+	j = (j + 1) % npcol;
     }
     for (k = 0; k < nsupers; ++k) {
 	knsupc = SuperSize( k );
