diff -ru freeyams.2011.09.23/sourceslib/blinde.c freeyams.2011.09.23-orig/sourceslib/blinde.c
--- freeyams.2011.09.23/sourceslib/blinde.c	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/blinde.c	2011-11-16 09:59:50.000000000 +0100
@@ -36,7 +36,7 @@
   ida =                          id[0]*10 + id[1];
   /*printf("date = %d %d %d\n",iy,im,ida); */
   if ( YMD(iy,im,ida) > YMD(MAX_YEAR,MAX_MONTH,MAX_DAY) ) {
-    fprintf(stdout,"  -- YAMS (LJLL), Version %s (%s)\n",VERSION,RELEASE);
+    fprintf(stdout,"  -- YAMS (LJLL), Version %s (%s)\n",YAMS_VERSION,RELEASE);
     fprintf(stdout,"     Copyright (C) LJLL, 1999-2006.\n\n");
     fprintf(stdout," ## Expiration date reached. Sorry.\n");
     fprintf(stdout,"    Please contact the author.\n");
diff -ru freeyams.2011.09.23/sourceslib/defines.h freeyams.2011.09.23-orig/sourceslib/defines.h
--- freeyams.2011.09.23/sourceslib/defines.h	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/defines.h	2011-11-16 09:59:50.000000000 +0100
@@ -18,7 +18,7 @@
 /* current version */
 #define DEFAULT_FILE    "DEFAULT.yams"
 #define RELEASE         "oct, 2006"
-#define VERSION         "2.4 b"
+#define YAMS_VERSION    "2.4 b"
 #define COPYRIGHT       "Copyright (C) LJLL, 1999-2006"
 
 #define REL    1
diff -ru freeyams.2011.09.23/sourceslib/extern.h freeyams.2011.09.23-orig/sourceslib/extern.h
--- freeyams.2011.09.23/sourceslib/extern.h	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/extern.h	2011-11-16 09:59:50.000000000 +0100
@@ -4,6 +4,7 @@
 
 
 #ifndef __YAMS
+#ifndef __YAMSLIB
   extern Error      yerr;
   extern Info       info;
   extern Options    opts;
@@ -14,7 +15,7 @@
   extern short     imprim;
   extern ubyte     ddebug;
 #endif
-
+#endif
 
 #ifdef __cplusplus
 }
diff -ru freeyams.2011.09.23/sourceslib/inout.c freeyams.2011.09.23-orig/sourceslib/inout.c
--- freeyams.2011.09.23/sourceslib/inout.c	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/inout.c	2011-11-16 09:59:50.000000000 +0100
@@ -581,7 +581,7 @@
     ppt->tag |= M_UNUSED;
     ppt->flag = ppt->color = 0;
   }
-
+  printf("sm->connex %d\n",sm->connex);
   if ( sm->connex > 0 ) {
     for (k=1; k<=sm->ne; k++) {
       pt1 = &sm->tria[k];
@@ -679,8 +679,8 @@
         natv++;
         if ( ppt->tag & M_CORNER )  tatv++;
       }
-      if ( !gs->new )  gs->new = ++nn;
-      if ( !gt->new )  gt->new = ++nt;
+      if ( !gs->newnum )  gs->newnum = ++nn;
+      if ( !gt->newnum )  gt->newnum = ++nt;
       if ( !pt1->edg[i] && pt1->tag[i] == M_NOTAG )  continue;
       else if ( pt1->adj[i] && (k > pt1->adj[i]) )   continue;
       nedge++;
@@ -864,12 +864,12 @@
   nn = nbl = 0;
   for (k=1; k<=sm->nvmax; k++) {
     gs = &sm->geom[k];
-    if ( gs->new > 0 ) {
+    if ( gs->newnum > 0 ) {
       iadr = nbl * 3;
       tabf[iadr+0] = gs->vn[0];
       tabf[iadr+1] = gs->vn[1];
       tabf[iadr+2] = gs->vn[2];
-      gs->new = ++nn;
+      gs->newnum = ++nn;
       ++nbl;
       if ( nbl == NMAX ) {
         LM_write_field(&ms, LM_Normals, nbl, tabf);
@@ -890,7 +890,7 @@
         iadr = nbl * 2;
         gs = &sm->geom[pt1->vn[i]];
         tabi[iadr+0] = ppt->tmp;
-        tabi[iadr+1] = gs->new;
+        tabi[iadr+1] = gs->newnum;
         ppt->flag = 1;
         ++nbl;
         if ( nbl == NMAX ) {
@@ -916,7 +916,7 @@
         iadr = nbl * 3;
         tabi[iadr+0] = nn;
         tabi[iadr+1] = i+1;
-        tabi[iadr+2] = gs->new;
+        tabi[iadr+2] = gs->newnum;
         ++nbl;
         if ( nbl == NMAX ) {
           LM_write_field(&ms, LM_NormalAtTriangleVertices, nbl, tabi);
@@ -931,12 +931,12 @@
   nt = nbl = 0;
   for (k=1; k<=sm->ntmax; k++) {
     gt = &sm->tgte[k];
-    if ( gt->new > 0 ) {
+    if ( gt->newnum > 0 ) {
       iadr = nbl * 3;
       tabf[iadr+0] = gt->t[0];
       tabf[iadr+1] = gt->t[1];
       tabf[iadr+2] = gt->t[2];
-      gt->new  = ++nt;
+      gt->newnum  = ++nt;
       ++nbl;
       if ( nbl == NMAX ) {
         LM_write_field(&ms, LM_Tangents, nbl, tabf);
@@ -956,7 +956,7 @@
       iadr = nbl * 2;
       gt = &sm->tgte[ppt->tge];
       tabi[iadr+0]  = ppt->tmp;
-      tabi[iadr+1]  = gt->new;
+      tabi[iadr+1]  = gt->newnum;
       ppt->flag = 1;
       ++nbl;
       if ( nbl == NMAX ) {
@@ -1105,8 +1105,8 @@
         natv++;
         if ( ppt->tag & M_CORNER )  tatv++;
       }
-      if ( !gs->new )  gs->new = ++nn;
-      if ( !gt->new )  gt->new = ++nq;
+      if ( !gs->newnum )  gs->newnum = ++nn;
+      if ( !gt->newnum )  gt->newnum = ++nq;
       if ( !pq1->edg[i] && pq1->tag[i] == M_NOTAG )  continue;
       else if ( pq1->adj[i] && (k > pq1->adj[i]) )   continue;
       nedge++;
@@ -1263,12 +1263,12 @@
   nn = nbl = 0;
   for (k=1; k<=sm->nvmax; k++) {
     gs = &sm->geom[k];
-    if ( gs->new > 0 ) {
+    if ( gs->newnum > 0 ) {
       iadr = nbl * 3;
       tabf[iadr+0] = gs->vn[0];
       tabf[iadr+1] = gs->vn[1];
       tabf[iadr+2] = gs->vn[2];
-      gs->new = ++nn;
+      gs->newnum = ++nn;
       ++nbl;
       if ( nbl == NMAX ) {
         LM_write_field(&ms, LM_Normals, nbl, tabf);
@@ -1289,7 +1289,7 @@
         iadr = nbl * 2;
         gs = &sm->geom[pq1->vn[i]];
         tabi[iadr+0] = ppt->tmp;
-        tabi[iadr+1] = gs->new;
+        tabi[iadr+1] = gs->newnum;
         ppt->flag = 1;
         ++nbl;
         if ( nbl == NMAX ) {
@@ -1315,7 +1315,7 @@
     iadr = nbl * 3;
     tabi[iadr+0] = nn;
     tabi[iadr+1] = i+1;
-    tabi[iadr+2] = gs->new;
+    tabi[iadr+2] = gs->newnum;
     ++nbl;
         if ( nbl == NMAX ) {
           LM_write_field(&ms, LM_NormalAtQuadrilateralVertices, nbl, tabi);
@@ -1330,12 +1330,12 @@
   nq = nbl = 0;
   for (k=1; k<=sm->ntmax; k++) {
     gt = &sm->tgte[k];
-    if ( gt->new > 0 ) {
+    if ( gt->newnum > 0 ) {
       iadr = nbl * 3;
       tabf[iadr+0] = gt->t[0];
       tabf[iadr+1] = gt->t[1];
       tabf[iadr+2] = gt->t[2];
-      gt->new  = ++nq;
+      gt->newnum  = ++nq;
       ++nbl;
       if ( nbl == NMAX ) {
         LM_write_field(&ms, LM_Tangents, nbl, tabf);
@@ -1355,7 +1355,7 @@
       iadr = nbl * 2;
       gt = &sm->tgte[ppt->tge];
       tabi[iadr+0]  = ppt->tmp;
-      tabi[iadr+1]  = gt->new;
+      tabi[iadr+1]  = gt->newnum;
       ppt->flag = 1;
       ++nbl;
       if ( nbl == NMAX ) {
@@ -1524,8 +1524,8 @@
   /* adjust sizes */
   if ( opts.hmin < 0.0 )
     opts.hmin = max(opts.hmin,hmin);
-    if ( opts.hmax < 0.0 )
-      opts.hmax = max(opts.hmax,hmax);
+  if ( opts.hmax < 0.0 )
+    opts.hmax = max(opts.hmax,hmax);
 
   E_pop();
   return(1);
Only in freeyams.2011.09.23-orig/sourceslib: inout.c.orig
diff -ru freeyams.2011.09.23/sourceslib/parsop.c freeyams.2011.09.23-orig/sourceslib/parsop.c
--- freeyams.2011.09.23/sourceslib/parsop.c	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/parsop.c	2011-11-16 09:59:50.000000000 +0100
@@ -16,7 +16,7 @@
   }
 
   /* create standard parameter file */
-  fprintf(out,"#  Generated by YAMS %s\n",VERSION);
+  fprintf(out,"#  Generated by YAMS %s\n",YAMS_VERSION);
   fprintf(out,"#  Uncomment next if absolute unit desired\n");
   fprintf(out,"#Absolute\n");
 
Only in freeyams.2011.09.23-orig/sourceslib: parsop.c.orig
diff -ru freeyams.2011.09.23/sourceslib/primsg.c freeyams.2011.09.23-orig/sourceslib/primsg.c
--- freeyams.2011.09.23/sourceslib/primsg.c	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/primsg.c	2011-11-16 09:59:50.000000000 +0100
@@ -18,10 +18,10 @@
     case 0000:
       if ( opts.ctrl & ISO )
         fprintf(stdout,"\n  %s\n   MODULE YAMS-LJLL : %s\n  %s\n",
-	        STR,VERSION,STR);
+	        STR,YAMS_VERSION,STR);
       else
         fprintf(stdout,"\n  %s\n   MODULE YAMS-LJLL (ANISO) : %s\n  %s\n",
-	        STR,VERSION,STR);
+	        STR,YAMS_VERSION,STR);
       return;
     case 0001: 
       fprintf(stdout,"\n  %s\n   END OF MODULE YAMS\n  %s\n\n",STR,STR);
diff -ru freeyams.2011.09.23/sourceslib/surf.h freeyams.2011.09.23-orig/sourceslib/surf.h
--- freeyams.2011.09.23/sourceslib/surf.h	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/surf.h	2011-11-16 09:59:50.000000000 +0100
@@ -64,13 +64,13 @@
 typedef struct geomsupp {
   float    vn[3];           /* array of vertex normals  */
   float    gap;             /* local gap value          */
-  int      new;             /* pointer to new number    */
+  int      newnum;             /* pointer to new number    */
 } GeomSupp;
 typedef GeomSupp  * pGeomSupp;
 
 typedef struct geomtge {
   float    t[3];
-  int      new;
+  int      newnum;
 } Geomtge;
 typedef Geomtge   * pGeomtge;
 
diff -ru freeyams.2011.09.23/sourceslib/yams.c freeyams.2011.09.23-orig/sourceslib/yams.c
--- freeyams.2011.09.23/sourceslib/yams.c	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/yams.c	2011-11-16 09:59:50.000000000 +0100
@@ -24,6 +24,7 @@
 ubyte     ddebug;
 ubyte     ecp;
 
+long verbosity; 
 
 static void excfun(int sigid) {
   fprintf(stdout,"\n Unexpected error:");  fflush(stdout);
Only in freeyams.2011.09.23-orig/sourceslib: yams.c.orig
diff -ru freeyams.2011.09.23/sourceslib/yams.h freeyams.2011.09.23-orig/sourceslib/yams.h
--- freeyams.2011.09.23/sourceslib/yams.h	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/yams.h	2011-11-16 09:59:50.000000000 +0100
@@ -6,12 +6,17 @@
 #include <string.h>
 
 #include "chrono.h"
-#include "edge.h"
-#include "error.h"
-#include "hash.h"
-#include "stack.h"
-#include "info.h"
 #include "memory.h"
-#include "option.h"
-#include "surf.h"
+
+//#include "edge.h"
+//#include "error.h"
+//#include "hash.h"
+//#include "stack.h"
+//#include "info.h"
+//#include "memory.h"
+//#include "option.h"
+//#include "surf.h"
+
+#include "yamslib.h"
+#include "yamslib_internal.h"
 #include "global.h"
diff -ru freeyams.2011.09.23/sourceslib/yams0.c freeyams.2011.09.23-orig/sourceslib/yams0.c
--- freeyams.2011.09.23/sourceslib/yams0.c	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/yams0.c	2011-11-16 09:59:50.000000000 +0100
@@ -24,7 +24,7 @@
   if ( imprim ) {
     if ( ecp )
       fprintf(stdout,"  ## Version:  Ecole Centrale de Paris ##\n\n");
-    fprintf(stdout,"  -- YAMS (LJLL) Version %s (%s)\n",VERSION,RELEASE);
+    fprintf(stdout,"  -- YAMS (LJLL) Version %s (%s)\n",YAMS_VERSION,RELEASE);
     fprintf(stdout,"     %s.\n",COPYRIGHT);
     fprintf(stdout,"     compiled: %s.\n\n",COMPIL);
   }
diff -ru freeyams.2011.09.23/sourceslib/zaldy1.c freeyams.2011.09.23-orig/sourceslib/zaldy1.c
--- freeyams.2011.09.23/sourceslib/zaldy1.c	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/zaldy1.c	2011-11-16 09:59:50.000000000 +0100
@@ -45,13 +45,20 @@
     }
   }
   else {
+    int bytes = sizeof(Point) \
+              + sizeof(Metric) \
+              + 2   * sizeof(Triangle) \
+              + 1.5 * sizeof(GeomSupp) \
+              + 0.2 * sizeof(Geomtge) \
+              + 3.7 * sizeof(Hashtable);
+    /*
     int bytes = sizeof(struct spoint) \
               + sizeof(struct metric) \
               + 2   * sizeof(struct striangle) \
               + 1.5 * sizeof(struct geomsupp) \
               + 0.2 * sizeof(struct geomtge) \
               + 3.7 * sizeof(struct shashtab);
-
+    */
     sm->npmax = (int)((double)memory / bytes * million);
     sm->npmax = max(sm->npmax,npmax);
     sm->nemax = max(2 * sm->npmax,nemax);
diff -ru freeyams.2011.09.23/sourceslib/zaldy3.c freeyams.2011.09.23-orig/sourceslib/zaldy3.c
--- freeyams.2011.09.23/sourceslib/zaldy3.c	2011-11-16 11:06:08.000000000 +0100
+++ freeyams.2011.09.23-orig/sourceslib/zaldy3.c	2011-11-16 09:59:50.000000000 +0100
@@ -34,7 +34,7 @@
     for (j=sm->nvmax; j<nvsize; j++) {
       sm->geom[j].vn[0] = sm->geom[j].vn[1] = sm->geom[j].vn[2] = 0.;
       sm->geom[j].gap   = 1.;
-      sm->geom[j].new   = 0;
+      sm->geom[j].newnum   = 0;
     }
     sm->nvmax = nvsize;
     yerr.inderr[0] = sm->nvmax;
diff -u freeyams.2012.02.05/sourceslib/outqua_a.c  freeyams.2011.09.23-orig/sourceslib/outqua_a.c
--- freeyams.2012.02.05/sourceslib/outqua_a.c	2009-05-13 21:00:03.000000000 +0200
+++ freeyams.2011.09.23-orig/sourceslib/outqua_a.c	2012-09-04 08:51:04.000000000 +0200
@@ -219,7 +219,7 @@
   pt = &mesh->tria[iel];
   fprintf(stdout,"     WORST ELEMENT   %d (%d)   %d %d %d\n",iel,ielr,pt->v[0],pt->v[1],pt->v[2]);
 
-  if ( abs(imprim) < 5 )  return;
+  if ( abs(imprim) < 5 )  return (1) ;
  
   fprintf(stdout,"\n     HISTOGRAMM\n");
   imax = min(9,(int)(10.*rapmax));
--- freeyams.2012.02.05/sourceslib/eigenv.c	2016-01-28 11:47:05.000000000 +0100
+++ freeyams.2012.02.05-orig/sources/eigenv.c	2012-02-04 12:58:11.000000000 +0100
@@ -3,7 +3,7 @@
 #include <math.h>
 
 /* seeking 1.e-05 accuracy */
+#define  EPSD           1.e-12
-#define  EPSD           1.e-15
 #define  EPSD2          1.e-200
 #define  EPS6           5.e-06
 #define  EPS            1.e-06
--- freeyams.2012.02.05/sourceslib/sproto.h	2021-02-17 11:09:34.000000000 +0100
+++ freeyams.2012.02.05-orig/sourceslib/sproto.h	2009-05-13 20:56:39.000000000 +0200
@@ -161,3 +161,19 @@
 int   outmsh_q(pSurfMesh ,char *);
 int   updtop_q(pSurfMesh );
 int   qualfa_q(float *,float *,float *,float *,float *,float *);
+
+int hippop(pTriangle tria);
+int updqua_a(pSurfMesh sm);
+int loptia3(pSurfMesh sm,ubyte fechk,float declic) ;
+int angdef(pSurfMesh sm);
+int boulep2(pSurfMesh sm,int depart,int i,Ball *b);
+int simred_a(double m1[6],double m2[6],double m[6]);
+int typelt(pSurfMesh sm,int k);
+int taspoi(pSurfMesh sm);
+int yams6(pSurfMesh sm);
+int yamsq(pSurfMesh sm);
+int yams22(pSurfMesh sm);
+int optedg(pSurfMesh sm) ;
+void primem(int np);
+#include "eigenv.h"
+#include <ctype.h>
--- freeyams.2012.02.05/sourceslib/error.c	2006-07-12 18:32:35.000000000 +0200
+++ freeyams.2012.02.05-orig/sourceslib/error.c	2021-02-17 10:58:12.000000000 +0100
@@ -10,7 +10,7 @@
 #endif
 
 #include <stdio.h>
-/*#include <ctype.h>*/
+#include <ctype.h>
 #include "string.h"
 
 #define MAXLEV   25
