diff -ur hips-1.2b-rc4/SRC/IO/io_hb.c hips-1.2b-rc4-okk/io_hb.c
--- hips-1.2b-rc4/SRC/IO/io_hb.c	2009-05-11 15:59:33.000000000 +0200
+++ hips-1.2b-rc4-okk/io_hb.c	2010-02-09 13:43:39.000000000 +0100
@@ -222,7 +222,11 @@
 #include<stdlib.h>
 #include<string.h>
 #include<math.h>
+#ifdef __APPLE__
+#include <malloc/malloc.h>
+#else
 #include<malloc.h>
+#endif
 #include <assert.h>
 
 /** macros FSCANF and FGETS **/

diff -ur hips-1.2b-rc4/SRC/IO/io_hb.h hips-1.2b-rc4-okk/io_hb.h
--- hips-1.2b-rc4/SRC/IO/io_hb.h	2009-02-23 23:59:03.000000000 +0100
+++ hips-1.2b-rc4-okk/io_hb.h	2010-02-09 12:22:32.000000000 +0100
@@ -5,7 +5,11 @@
 
 #include<stdio.h>
 #include<stdlib.h>
+#ifdef __APPLE__
+#include<malloc/malloc.h>
+#else
 #include<malloc.h>
+#endif
 #ifdef __cplusplus
 extern "C" {
 #endif

diff -ur hips-1.2b-rc4/SRC/IO/io_mm.c hips-1.2b-rc4-okk/io_mm.c
--- hips-1.2b-rc4/SRC/IO/io_mm.c	2009-05-11 15:59:33.000000000 +0200
+++ hips-1.2b-rc4-okk/io_mm.c	2010-02-09 13:56:20.000000000 +0100
@@ -11,7 +11,11 @@
 
 #include <stdio.h>
 #include <string.h>
+#ifdef __APPLE__
+#include <malloc/malloc.h>
+#else
 #include <malloc.h>
+#endif
 #include <ctype.h>
