diff -ur hips-1.2b-rc4/SRC/INCLUDE/hips.h hips-1.2b-rc4-okk/SRC/INCLUDE/hips.h
--- hips-1.2b-rc4/SRC/INCLUDE/hips.h	2009-06-21 23:12:59.000000000 +0200
+++ hips-1.2b-rc4-okk/SRC/INCLUDE/hips.h	2010-06-17 10:42:14.000000000 +0200
@@ -122,6 +122,7 @@
 #ifndef HIPS_H
 #define HIPS_H
 #include <stdio.h>
+#include <ctype.h>
 #include "type.h"
 
 #ifdef INTL
diff -ur hips-1.2b-rc4/SRC/INCLUDE/makefile hips-1.2b-rc4-okk/SRC/INCLUDE/makefile
--- hips-1.2b-rc4/SRC/INCLUDE/makefile	2009-03-09 13:34:49.000000000 +0100
+++ hips-1.2b-rc4-okk/SRC/INCLUDE/makefile	2010-06-17 10:42:14.000000000 +0200
@@ -7,7 +7,7 @@
 	$(CP) -f hips.inc $(HIPS_DIR)/LIB/
 
 hips.inc:  hips_interface.F90
-	$(CC) $(CC_OPT) -E hips_interface.F90 > hips.inc
+	$(CPP) $(CFLAGS) $(CC_OPT) hips_interface.F90 > hips.inc
 
 .h:
 	$(CP) -f $< $(HIPS_DIR)/LIB/
Only in hips-1.2b-rc4-okk/SRC/INCLUDE: makefile.orig
diff -ur hips-1.2b-rc4/SRC/INCLUDE/queue.h hips-1.2b-rc4-okk/SRC/INCLUDE/queue.h
--- hips-1.2b-rc4/SRC/INCLUDE/queue.h	2009-06-21 23:12:57.000000000 +0200
+++ hips-1.2b-rc4-okk/SRC/INCLUDE/queue.h	2010-06-25 08:21:50.000000000 +0200
@@ -98,7 +98,7 @@
 } Heap;
 
 
-#define static
+/*#define static*/
 
 int     Heap_Init       (Heap *, dim_t size);
 void    Heap_Exit       (Heap *);
diff -ur hips-1.2b-rc4/SRC/IO/io_hb.c hips-1.2b-rc4-okk/SRC/IO/io_hb.c
--- hips-1.2b-rc4/SRC/IO/io_hb.c	2009-05-11 15:59:33.000000000 +0200
+++ hips-1.2b-rc4-okk/SRC/IO/io_hb.c	2010-06-17 10:42:14.000000000 +0200
@@ -222,7 +222,11 @@
 #include<stdlib.h>
 #include<string.h>
 #include<math.h>
+#ifdef __APPLE__
+#include <malloc/malloc.h>
+#else
 #include<malloc.h>
+#endif
 #include <assert.h>
 
 /** macros FSCANF and FGETS **/
diff -ur hips-1.2b-rc4/SRC/IO/io_hb.h hips-1.2b-rc4-okk/SRC/IO/io_hb.h
--- hips-1.2b-rc4/SRC/IO/io_hb.h	2009-02-23 23:59:03.000000000 +0100
+++ hips-1.2b-rc4-okk/SRC/IO/io_hb.h	2010-06-17 10:42:14.000000000 +0200
@@ -5,7 +5,11 @@
 
 #include<stdio.h>
 #include<stdlib.h>
+#ifdef __APPLE__
+#include<malloc/malloc.h>
+#else
 #include<malloc.h>
+#endif
 #ifdef __cplusplus
 extern "C" {
 #endif
diff -ur hips-1.2b-rc4/SRC/IO/io_mm.c hips-1.2b-rc4-okk/SRC/IO/io_mm.c
--- hips-1.2b-rc4/SRC/IO/io_mm.c	2009-05-11 15:59:33.000000000 +0200
+++ hips-1.2b-rc4-okk/SRC/IO/io_mm.c	2010-06-17 10:42:14.000000000 +0200
@@ -11,7 +11,11 @@
 
 #include <stdio.h>
 #include <string.h>
+#ifdef __APPLE__
+#include <malloc/malloc.h>
+#else
 #include <malloc.h>
+#endif
 #include <ctype.h>
 
 
diff -ur hips-1.2b-rc4/SRC/IO/makefile hips-1.2b-rc4-okk/SRC/IO/makefile
--- hips-1.2b-rc4/SRC/IO/makefile	2009-03-19 18:13:16.000000000 +0100
+++ hips-1.2b-rc4-okk/SRC/IO/makefile	2010-06-17 10:42:14.000000000 +0200
@@ -17,7 +17,7 @@
 	$(AR) $(ARFLAGS) $(LIBIOMM) $(OBJ) 
 
 uread.f90: template-uread.F90
-	$(CC) $(CFLAGS) $(CC_OPT) -E $< > $@
+	$(CPP) $(CFLAGS) $(CC_OPT) $< > $@
 
 %.o: %.c
 	$(CC) $(CC_OPT) $< -c -o $@
Only in hips-1.2b-rc4-okk/SRC/IO: makefile.orig
diff -ur hips-1.2b-rc4/SRC/SPKIT/makefile hips-1.2b-rc4-okk/SRC/SPKIT/makefile
--- hips-1.2b-rc4/SRC/SPKIT/makefile	2009-03-19 18:13:16.000000000 +0100
+++ hips-1.2b-rc4-okk/SRC/SPKIT/makefile	2010-06-17 10:42:14.000000000 +0200
@@ -6,6 +6,7 @@
 	OBJ   = $(TMP:.f90=.o)
 else
 	OBJ   = 
+	AR    = echo do nothing : ar 
 endif
 
 default: $(OBJ)
@@ -15,10 +16,10 @@
 	$(FC) $(FC_OPT) $< -c -o $@
 
 skitfc.f90: template-skitfc.F90
-	$(CC) $(CFLAGS) $(CC_OPT) -E $< > $@
+	$(CPP) $(CFLAGS) $(CC_OPT) $< > $@
 
 skitf.f90: template-skitf.F90
-	$(CC) $(CFLAGS) $(CC_OPT) -E $< > $@
+	$(CPP) $(CFLAGS) $(CC_OPT) $< > $@
 
 clean:
 	rm -f $(TMP) *.o *.ex* *core ./extras/core* out* \#*  *~
Only in hips-1.2b-rc4-okk/SRC/SPKIT: makefile.orig
diff -ur hips-1.2b-rc4/TESTS/PARALLEL/makefile hips-1.2b-rc4-okk/TESTS/PARALLEL/makefile
--- hips-1.2b-rc4/TESTS/PARALLEL/makefile	2009-04-30 10:24:48.000000000 +0200
+++ hips-1.2b-rc4-okk/TESTS/PARALLEL/makefile	2010-06-17 10:42:14.000000000 +0200
@@ -41,17 +41,17 @@
 	$(MPICC) $(MPICC_OPT) $< -c -o $@
 
 testHIPS1-Fortran.f90 : ../../SRC/INCLUDE/hips.inc templateHIPS1-Fortran.F90
-	$(CC) $(CFLAGS) $(CC_OPT) -E  templateHIPS1-Fortran.F90 > testHIPS1-Fortran.f90
+	$(CPP) $(CFLAGS) $(CC_OPT)  templateHIPS1-Fortran.F90 > testHIPS1-Fortran.f90
 
 testHIPS2-Fortran.f90 : ../../SRC/INCLUDE/hips.inc templateHIPS2-Fortran.F90
-	$(CC) $(CFLAGS) $(CC_OPT) -E  templateHIPS2-Fortran.F90 > testHIPS2-Fortran.f90
+	$(CPP) $(CFLAGS) $(CC_OPT)  templateHIPS2-Fortran.F90 > testHIPS2-Fortran.f90
 
 
 testHIPS3-Fortran.f90 : ../../SRC/INCLUDE/hips.inc templateHIPS3-Fortran.F90
-	$(CC) $(CFLAGS) $(CC_OPT) -E  templateHIPS3-Fortran.F90 > testHIPS3-Fortran.f90
+	$(CPP) $(CFLAGS) $(CC_OPT)  templateHIPS3-Fortran.F90 > testHIPS3-Fortran.f90
 
 testHIPS-Laplace1-Fortran.f90 : ../../SRC/INCLUDE/hips.inc templateHIPS-Laplace1-Fortran.F90
-	$(CC) $(CFLAGS) $(CC_OPT) -E  templateHIPS-Laplace1-Fortran.F90 > testHIPS-Laplace1-Fortran.f90
+	$(CPP) $(CFLAGS) $(CC_OPT)  templateHIPS-Laplace1-Fortran.F90 > testHIPS-Laplace1-Fortran.f90
 
 testHIPS1-Fortran.o: testHIPS1-Fortran.f90 $(LIBHIPS)
 	$(MPIFC) $(MPIFC_OPT) $< -c -o $@
Only in hips-1.2b-rc4-okk/TESTS/PARALLEL: makefile.orig
Only in hips-1.2b-rc4-okk: makefile.inc
Only in hips-1.2b-rc4-okk: tag-ff++
