diff -ur tetgen1.4.2/tetgen.cxx tetgen1.4.2-okk/tetgen.cxx
--- tetgen1.4.2/tetgen.cxx	2007-04-16 16:45:11.000000000 +0200
+++ tetgen1.4.2-okk/tetgen.cxx	2009-05-25 10:37:19.000000000 +0200
@@ -3506,7 +3506,7 @@
 //                                                                           //
 ///////////////////////////////////////////////////////////////////////////////
 
-void tetgenmesh::set_compfunc(char* str, int* itbytes, compfunc* pcomp)
+void tetgenmesh::set_compfunc(const char* str, int* itbytes, compfunc* pcomp)
 {
   // First figure out whether it is a pointer or not.
   if (str[strlen(str) - 1] == '*') {
@@ -25978,6 +25978,7 @@
   }
   delete [] oldtetlist;
   delete [] newtetlist;
+  delete [] newpt; // Add by J. Morice for FreeFem++ (missing delete) Nov 2008
   // Clear work lists.
   newsegshlist->clear();
   spinshlist->clear();
diff -ur tetgen1.4.2/tetgen.h tetgen1.4.2-okk/tetgen.h
--- tetgen1.4.2/tetgen.h	2007-04-16 16:45:13.000000000 +0200
+++ tetgen1.4.2-okk/tetgen.h	2009-05-25 10:37:41.000000000 +0200
@@ -980,7 +980,7 @@
 
     // The function used to determine the size of primitive data types and
     //   set the corresponding predefined linear order functions for them.
-    static void set_compfunc(char* str, int* itembytes, compfunc* pcomp);
+    static void set_compfunc(const char* str, int* itembytes, compfunc* pcomp);
 
 ///////////////////////////////////////////////////////////////////////////////
 //                                                                           //
@@ -1021,7 +1021,7 @@
         list(int itbytes, compfunc pcomp, int mitems = 256, int exsize = 128) {
           listinit(itbytes, pcomp, mitems, exsize);
         }
-        list(char* str, int mitems = 256, int exsize = 128) {
+        list(const char* str, int mitems = 256, int exsize = 128) {
           set_compfunc(str, &itembytes, &comp);
           listinit(itembytes, comp, mitems, exsize);
         }
