--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/blacs_get_.c	2011-09-28 19:07:24.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/blacs_get_.c	2019-05-07 16:14:28.000000000 +0200
@@ -23,7 +23,7 @@
    case SGET_MSGIDS:
       if (BI_COMM_WORLD == NULL) Cblacs_pinfo(val, &val[1]);
       iptr = &val[1];
-      ierr=MPI_Attr_get(MPI_COMM_WORLD, MPI_TAG_UB, (BVOID **) &iptr,val);
+      ierr=MPI_Comm_get_attr(MPI_COMM_WORLD, MPI_TAG_UB, (BVOID **) &iptr,val);
       val[0] = 0;
       val[1] = *iptr;
       break;
--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/cgamn2d_.c	2012-04-24 18:26:51.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/cgamn2d_.c	2019-05-07 16:14:28.000000000 +0200
@@ -221,7 +221,7 @@
       {
 #endif
       i = 2;
-      ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
+      ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
       ierr=MPI_Type_commit(&MyType);
       bp->N = bp2->N = 1;
       bp->dtype = bp2->dtype = MyType;
--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/cgamx2d_.c	2012-04-24 18:26:51.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/cgamx2d_.c	2019-05-07 16:14:28.000000000 +0200
@@ -221,7 +221,7 @@
       {
 #endif
       i = 2;
-      ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
+      ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
       ierr=MPI_Type_commit(&MyType);
       bp->N = bp2->N = 1;
       bp->dtype = bp2->dtype = MyType;
--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/dgamn2d_.c	2012-04-24 18:26:51.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/dgamn2d_.c	2019-05-07 16:14:28.000000000 +0200
@@ -221,7 +221,7 @@
       {
 #endif
       i = 2;
-      ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
+      ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
       ierr=MPI_Type_commit(&MyType);
       bp->N = bp2->N = 1;
       bp->dtype = bp2->dtype = MyType;
--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/dgamx2d_.c	2012-04-24 18:26:51.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/dgamx2d_.c	2019-05-07 16:14:28.000000000 +0200
@@ -221,7 +221,7 @@
       {
 #endif
       i = 2;
-      ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
+      ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
       ierr=MPI_Type_commit(&MyType);
       bp->N = bp2->N = 1;
       bp->dtype = bp2->dtype = MyType;
--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/igamn2d_.c	2012-04-24 18:26:51.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/igamn2d_.c	2019-05-07 16:14:28.000000000 +0200
@@ -218,7 +218,7 @@
       {
 #endif
       i = 2;
-      ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
+      ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
       ierr=MPI_Type_commit(&MyType);
       bp->N = bp2->N = 1;
       bp->dtype = bp2->dtype = MyType;
--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/igamx2d_.c	2012-04-24 18:26:51.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/igamx2d_.c	2019-05-07 16:14:28.000000000 +0200
@@ -218,7 +218,7 @@
       {
 #endif
       i = 2;
-      ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
+      ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
       ierr=MPI_Type_commit(&MyType);
       bp->N = bp2->N = 1;
       bp->dtype = bp2->dtype = MyType;
--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/sgamn2d_.c	2012-04-24 18:26:51.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/sgamn2d_.c	2019-05-07 16:14:28.000000000 +0200
@@ -221,7 +221,7 @@
       {
 #endif
       i = 2;
-      ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
+      ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
       ierr=MPI_Type_commit(&MyType);
       bp->N = bp2->N = 1;
       bp->dtype = bp2->dtype = MyType;
--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/sgamx2d_.c	2012-04-24 18:26:51.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/sgamx2d_.c	2019-05-07 16:14:28.000000000 +0200
@@ -221,7 +221,7 @@
       {
 #endif
       i = 2;
-      ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
+      ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
       ierr=MPI_Type_commit(&MyType);
       bp->N = bp2->N = 1;
       bp->dtype = bp2->dtype = MyType;
--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/zgamn2d_.c	2012-04-24 18:26:51.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/zgamn2d_.c	2019-05-07 16:14:28.000000000 +0200
@@ -221,7 +221,7 @@
       {
 #endif
       i = 2;
-      ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
+      ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
       ierr=MPI_Type_commit(&MyType);
       bp->N = bp2->N = 1;
       bp->dtype = bp2->dtype = MyType;
--- /Users/hecht/ff/ff-test/3rdparty/scalapack/scalapack-2.0.2/BLACS/SRC/zgamx2d_.c	2012-04-24 18:26:51.000000000 +0200
+++ /Users/hecht/ff/ff-git/3rdparty/ff-petsc/petsc-3.11.1/ff-real/externalpackages/git.scalapack/BLACS/SRC/zgamx2d_.c	2019-05-07 16:14:28.000000000 +0200
@@ -221,7 +221,7 @@
       {
 #endif
       i = 2;
-      ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
+      ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
       ierr=MPI_Type_commit(&MyType);
       bp->N = bp2->N = 1;
       bp->dtype = bp2->dtype = MyType;
