diff -ur scotch_5.1_esmumps/src/libscotch/Makefile scotch_5.1_esmumps-okk/Makefile
--- scotch_5.1_esmumps/src/libscotch/Makefile	2009-06-06 16:05:52.000000000 +0200
+++ scotch_5.1_esmumps-okk/Makefile	2010-02-11 18:14:35.000000000 +0100
@@ -64,10 +64,9 @@
 					scotch.h									\
 					scotchf.h									\
 					libptscotch$(LIB)								\
-					libscotch$(LIB)									\
 					libptscotcherr$(LIB)								\
 					libptscotcherrexit$(LIB)
-
+					## libscotch$(LIB)	
 install				:
 					-$(CP) scotch.h scotchf.h $(includedir)
 					-$(CP) libscotch$(LIB) libscotcherr*$(LIB) $(libdir)
@@ -2274,7 +2273,162 @@
 					vdgraph_separate_sq$(OBJ)		\
 					vdgraph_separate_st$(OBJ)		\
 					vdgraph_separate_zr$(OBJ)		\
-					vdgraph_store$(OBJ)
+					vdgraph_store$(OBJ) arch$(OBJ) 		\
+					arch_build$(OBJ)			\
+					arch_cmplt$(OBJ)			\
+					arch_cmpltw$(OBJ)			\
+					arch_deco$(OBJ)				\
+					arch_hcub$(OBJ)				\
+					arch_mesh$(OBJ)				\
+					arch_tleaf$(OBJ)			\
+					arch_torus$(OBJ)			\
+					arch_vcmplt$(OBJ)			\
+					arch_vhcub$(OBJ)			\
+					bgraph$(OBJ)				\
+					bgraph_bipart_bd$(OBJ)			\
+					bgraph_bipart_df$(OBJ)			\
+					bgraph_bipart_ex$(OBJ)			\
+					bgraph_bipart_fm$(OBJ)			\
+					bgraph_bipart_gg$(OBJ)			\
+					bgraph_bipart_gp$(OBJ)			\
+					bgraph_bipart_ml$(OBJ)			\
+					bgraph_bipart_st$(OBJ)			\
+					bgraph_bipart_zr$(OBJ)			\
+					bgraph_check$(OBJ)			\
+					bgraph_store$(OBJ)			\
+					common$(OBJ)				\
+					common_file$(OBJ)			\
+					common_file_compress$(OBJ)		\
+					common_file_uncompress$(OBJ)		\
+					common_integer$(OBJ)			\
+					common_memory$(OBJ)			\
+					common_stub$(OBJ)			\
+					gain$(OBJ)				\
+					geom$(OBJ)				\
+					graph$(OBJ)				\
+					graph_base$(OBJ)			\
+					graph_check$(OBJ)			\
+					graph_coarsen$(OBJ)			\
+					graph_induce$(OBJ)			\
+					graph_io$(OBJ)				\
+					graph_io_chac$(OBJ)			\
+					graph_io_habo$(OBJ)			\
+					graph_io_mmkt$(OBJ)			\
+					graph_io_scot$(OBJ)			\
+					graph_list$(OBJ)			\
+					hall_order_hd$(OBJ)			\
+					hall_order_hf$(OBJ)			\
+					hall_order_hx$(OBJ)			\
+					hgraph$(OBJ)				\
+					hgraph_check$(OBJ)			\
+					hgraph_induce$(OBJ)			\
+					hgraph_order_bl$(OBJ)			\
+					hgraph_order_cp$(OBJ)			\
+					hgraph_order_gp$(OBJ)			\
+					hgraph_order_hd$(OBJ)			\
+					hgraph_order_hf$(OBJ)			\
+					hgraph_order_hx$(OBJ)			\
+					hgraph_order_nd$(OBJ)			\
+					hgraph_order_si$(OBJ)			\
+					hgraph_order_st$(OBJ)			\
+					hmesh$(OBJ)				\
+					hmesh_check$(OBJ)			\
+					hmesh_hgraph$(OBJ)			\
+					hmesh_induce$(OBJ)			\
+					hmesh_mesh$(OBJ)			\
+					hmesh_order_bl$(OBJ)			\
+					hmesh_order_cp$(OBJ)			\
+					hmesh_order_gr$(OBJ)			\
+					hmesh_order_gp$(OBJ)			\
+					hmesh_order_hd$(OBJ)			\
+					hmesh_order_hf$(OBJ)			\
+					hmesh_order_hx$(OBJ)			\
+					hmesh_order_nd$(OBJ)			\
+					hmesh_order_si$(OBJ)			\
+					hmesh_order_st$(OBJ)			\
+					kgraph$(OBJ)				\
+					kgraph_map_rb$(OBJ)			\
+					kgraph_map_rb_map$(OBJ)			\
+					kgraph_map_rb_part$(OBJ)		\
+					kgraph_map_st$(OBJ)			\
+					library_arch$(OBJ)			\
+					library_arch_f$(OBJ)			\
+					library_arch_build$(OBJ)		\
+					library_arch_build_f$(OBJ)		\
+					library_geom$(OBJ)			\
+					library_geom_f$(OBJ)			\
+					library_graph$(OBJ)			\
+					library_graph_f$(OBJ)			\
+					library_graph_base$(OBJ)		\
+					library_graph_base_f$(OBJ)		\
+					library_graph_check$(OBJ)		\
+					library_graph_check_f$(OBJ)		\
+					library_graph_io_chac$(OBJ)		\
+					library_graph_io_chac_f$(OBJ)		\
+					library_graph_io_habo$(OBJ)		\
+					library_graph_io_habo_f$(OBJ)		\
+					library_graph_io_mmkt$(OBJ)		\
+					library_graph_io_mmkt_f$(OBJ)		\
+					library_graph_io_scot$(OBJ)		\
+					library_graph_io_scot_f$(OBJ)		\
+					library_graph_map$(OBJ)			\
+					library_graph_map_f$(OBJ)		\
+					library_graph_map_view$(OBJ)		\
+					library_graph_map_view_f$(OBJ)		\
+					library_graph_order$(OBJ)		\
+					library_graph_order_f$(OBJ)		\
+					library_mesh$(OBJ)			\
+					library_mesh_f$(OBJ)			\
+					library_mesh_graph$(OBJ)		\
+					library_mesh_graph_f$(OBJ)		\
+					library_mesh_io_habo$(OBJ)		\
+					library_mesh_io_habo_f$(OBJ)		\
+					library_mesh_io_scot$(OBJ)		\
+					library_mesh_io_scot_f$(OBJ)		\
+					library_mesh_order$(OBJ)		\
+					library_mesh_order_f$(OBJ)		\
+					library_parser$(OBJ)			\
+					library_parser_f$(OBJ)			\
+					library_random$(OBJ)			\
+					library_random_f$(OBJ)			\
+					mapping$(OBJ)				\
+					mapping_io$(OBJ)			\
+					mesh$(OBJ)				\
+					mesh_check$(OBJ)			\
+					mesh_coarsen$(OBJ)			\
+					mesh_graph$(OBJ)			\
+					mesh_induce_sepa$(OBJ)			\
+					mesh_io$(OBJ)				\
+					mesh_io_habo$(OBJ)			\
+					mesh_io_scot$(OBJ)			\
+					order$(OBJ)				\
+					order_check$(OBJ)			\
+					order_io$(OBJ)				\
+					parser$(OBJ)				\
+					parser_ll$(OBJ)				\
+					parser_yy$(OBJ)				\
+					vgraph$(OBJ)				\
+					vgraph_check$(OBJ)			\
+					vgraph_separate_bd$(OBJ)		\
+					vgraph_separate_es$(OBJ)		\
+					vgraph_separate_fm$(OBJ)		\
+					vgraph_separate_gg$(OBJ)		\
+					vgraph_separate_gp$(OBJ)		\
+					vgraph_separate_ml$(OBJ)		\
+					vgraph_separate_st$(OBJ)		\
+					vgraph_separate_th$(OBJ)		\
+					vgraph_separate_vw$(OBJ)		\
+					vgraph_separate_zr$(OBJ)		\
+					vgraph_store$(OBJ)			\
+					vmesh$(OBJ)				\
+					vmesh_check$(OBJ)			\
+					vmesh_separate_fm$(OBJ)			\
+					vmesh_separate_gg$(OBJ)			\
+					vmesh_separate_gr$(OBJ)			\
+					vmesh_separate_ml$(OBJ)			\
+					vmesh_separate_zr$(OBJ)			\
+					vmesh_separate_st$(OBJ)			\
+					vmesh_store$(OBJ)
 					$(AR) $(ARFLAGS) $(@) $(?)
 					-$(RANLIB) $(@)
 
