
--- pastix_release_2200/src/all_macros 2009-02-19 07:33:00.000000000 -0600
+++ pastix_release_2200/src/all_macros.new 2013-03-24 10:34:37.026406767 -0500
@@ -11,7 +11,7 @@
 OBJ		= $(OBJEXT)
 
 AR		= $(ARPROG)
-CCINC           = -I. -I$(DSTDIR) -I$(INSDIR)
+CCINC           = -I$(DSTDIR) -I$(INSDIR)
 CAT		= cat
 CP		= cp
 MKDIR		= mkdir
@@ -21,10 +21,10 @@
 TOUCH		= touch
 TAIL		= tail
 HEAD		= head
-VERSION         = `../../myversion.sh`
+VERSION         = ../../myversion.sh
 
 COMMONPASTIX_H	= 	$(INSDIR)/common_pastix.h	\
 			$(INSDIR)/api.h			\
 			$(INSDIR)/debug.h		\
 			$(INSDIR)/errors.h		\
-			$(INSDIR)/redefine_functions.h
\ No newline at end of file
+			$(INSDIR)/redefine_functions.h
