diff -ur pastix_release_2200/src/blend/src/make_gen pastix_release_2200-okk/blend/make_gen
--- pastix_release_2200/src/blend/src/make_gen	2009-02-19 14:33:00.000000000 +0100
+++ pastix_release_2200-okk/blend/make_gen	2010-02-12 11:03:07.000000000 +0100
@@ -90,8 +90,7 @@
 ##
 
 
-$(DSTDIR)/libblend3d$(LIB):		$(HEADERS) 			\
-					$(OJTDIR)/task$(OBJ)   		\
+$(DSTDIR)/libblend3d$(LIB):		$(OJTDIR)/task$(OBJ)   		\
 					$(OJTDIR)/fanboth2$(OBJ)   	\
 					$(OJTDIR)/splitfunc$(OBJ)   	\
 					$(OJTDIR)/symbol_cost$(OBJ)   	\
