--- Makefile	2019-07-04 09:11:55.000000000 +0200
+++ Makefile_windows	2019-07-04 09:11:53.000000000 +0200
@@ -127,20 +127,18 @@
 $(SRCDIR)/tag-conf-real:$(SRCDIR)/tag-tar
 	cd $(SRCDIR) &&  $(PYTHON2) ./configure MAKEFLAGS='' \
 		--prefix=$(DIR_INSTALL_REAL) \
-		--with-debugging=no \
-		--with-cc='$(MPICC)' --with-cxx='$(MPICXX)' --with-fc='$(MPIFC)' \
-		COPTFLAGS='-O3  $(FLAGS_MTUNE)' \
-		CXXOPTFLAGS='-O3  $(FLAGS_MTUNE)' \
-		FOPTFLAGS='-O3  $(FLAGS_MTUNE)' \
+		--with-mpi-lib=['/C/Program\ Files\ \(x86\)/Microsoft\ SDKs/MPI/Lib/x64/msmpi.lib'] \
+		--with-mpi-include=/C/Program\ Files\ \(x86\)/Microsoft\ SDKs/MPI/Include \
+		--with-fc=0 \
+		--download-metis \
+		--download-suitesparse \
+		--with-ssl=0 --with-x=0 --with-debugging=0  \
+		--download-f2cblaslapack=1 \
+		--with-mpiexec='/C/Program\ Files/Microsoft\ MPI/Bin/mpiexec'\
+		--download-slepc \
 		CFLAGS=' $(FLAGS_MTUNE)' \
 		CXXFLAGS='$(FLAGS_MTUNE)' \
 		FFLAGS='$(FLAGS_MTUNE)' \
-		--with-ssl=0 --with-x=0 --with-fortran-bindings=0 \
-		--with-scalar-type=real $(WITH_MPI_DIR) $(PETSC_ADDCONF) \
-		--with-blas-lapack-lib='$(BLAS_LAPACK_LIBS)' \
-		--download-scalapack --download-metis --download-ptscotch \
-		--download-mumps --download-ml --download-hypre --download-parmetis \
-		--download-superlu --download-suitesparse --download-fftw --download-slepc \
 		PETSC_ARCH=ff-real
 	test -f $(SRCDIR)/ff-real/lib/petsc/conf/petscvariables
 	touch $@
@@ -152,25 +150,19 @@
 $(SRCDIR)/tag-conf-complex:$(SRCDIR)/tag-tar Make-petsc-download.mk
 	cd $(SRCDIR) &&  $(PYTHON2) ./configure MAKEFLAGS='' \
 		--prefix=$(DIR_INSTALL_COMPLEX) \
-		--with-debugging=no \
-		--with-cc='$(MPICC)' --with-cxx='$(MPICXX)' --with-fc='$(MPIFC)' \
-		COPTFLAGS='-O3  $(FLAGS_MTUNE)' \
-		CXXOPTFLAGS='-O3  $(FLAGS_MTUNE)' \
-		FOPTFLAGS='-O3  $(FLAGS_MTUNE)' \
-		CFLAGS=' $(FLAGS_MTUNE)' \
-		CXXFLAGS='$(FLAGS_MTUNE)' \
-		FFLAGS='$(FLAGS_MTUNE)' \
+		--with-mpi-lib=['/C/Program\ Files\ \(x86\)/Microsoft\ SDKs/MPI/Lib/x64/msmpi.lib'] \
+		--with-mpi-include=/C/Program\ Files\ \(x86\)/Microsoft\ SDKs/MPI/Include \
+		--with-fc=0 \
 		--with-scalar-type=complex \
-		--with-ssl=0 --with-x=0 --with-fortran-bindings=0 $(WITH_MPI_DIR) $(PETSC_ADDCONF) \
-		--with-blas-lapack-lib='$(BLAS_LAPACK_LIBS)' \
 		--with-scalapack-dir=$(DIR_INSTALL_REAL) \
 		--with-metis-dir=$(DIR_INSTALL_REAL) \
-		--with-ptscotch-dir=$(DIR_INSTALL_REAL) \
-		--with-suitesparse-include='$(SUITESPARSE_INCLUDE)' --with-suitesparse-lib='$(SUITESPARSE_LIB)' \
-		--with-mumps-dir=$(DIR_INSTALL_REAL) \
-		--with-parmetis-dir=$(DIR_INSTALL_REAL) \
-		--with-fftw-dir=$(DIR_INSTALL_REAL) \
+		--with-ssl=0 --with-x=0 --with-debugging=0  
+		--with-blas-lapack-lib='$(BLAS_LAPACK_LIBS)' \ 
+		--with-mpiexec='/C/Program\ Files/Microsoft\ MPI/Bin/mpiexec' \
 		--download-slepc \
+		CFLAGS=' $(FLAGS_MTUNE)' \
+		CXXFLAGS='$(FLAGS_MTUNE)' \
+		FFLAGS='$(FLAGS_MTUNE)' \
 		PETSC_ARCH=ff-complex
 	test -f $(SRCDIR)/ff-complex/lib/petsc/conf/petscvariables
 	touch $@
