diff --git a/config/BuildSystem/config/packages/metis.py b/config/BuildSystem/config/packages/metis.py
index 136526da05..2d1c4d3271 100644
--- a/config/BuildSystem/config/packages/metis.py
+++ b/config/BuildSystem/config/packages/metis.py
@@ -44,6 +44,12 @@ class Configure(config.package.CMakePackage):
     if self.framework.argDB['download-metis-use-doubleprecision']:
       args.append('-DMETIS_USE_DOUBLEPRECISION=1')
     args.append('-DMATH_LIB="'+self.libraries.toStringNoDupes(self.mathlib.lib)+'"')
+    mpicc = self.framework.getMakeMacro('MPICC_SHOW')
+    mpicxx = self.framework.getMakeMacro('MPICXX_SHOW')
+    if mpicc and mpicxx:
+        args = self.rmArgsStartsWith(args,['-DCMAKE_CXX_COMPILER','-DCMAKE_C_COMPILER'])
+        args.append('-DCMAKE_C_COMPILER="'+mpicc.split(None, 1)[0]+'"')
+        args.append('-DCMAKE_CXX_COMPILER="'+mpicxx.split(None, 1)[0]+'"')
     return args

   def configureLibrary(self):
diff --git a/config/BuildSystem/config/packages/mmg.py b/config/BuildSystem/config/packages/mmg.py
index 1cc9122f41..4affbce3b3 100644
--- a/config/BuildSystem/config/packages/mmg.py
+++ b/config/BuildSystem/config/packages/mmg.py
@@ -27,6 +27,12 @@ class Configure(config.package.CMakePackage):
     args.append('-DUSE_ELAS=OFF')
     args.append('-DUSE_VTK=OFF')
     args.append('-DSCOTCH_DIR:STRING="'+self.ptscotch.directory+'"')
+    mpicc = self.framework.getMakeMacro('MPICC_SHOW')
+    mpicxx = self.framework.getMakeMacro('MPICXX_SHOW')
+    if mpicc and mpicxx:
+        args = self.rmArgsStartsWith(args,['-DCMAKE_CXX_COMPILER','-DCMAKE_C_COMPILER'])
+        args.append('-DCMAKE_C_COMPILER="'+mpicc.split(None, 1)[0]+'"')
+        args.append('-DCMAKE_CXX_COMPILER="'+mpicxx.split(None, 1)[0]+'"')
     return args

   def configureLibrary(self):
diff --git a/config/BuildSystem/config/packages/htool.py b/config/BuildSystem/config/packages/htool.py
index e6d07d15e0..f8d948ce37 100644
--- a/config/BuildSystem/config/packages/htool.py
+++ b/config/BuildSystem/config/packages/htool.py
@@ -3,7 +3,7 @@ import config.package
 class Configure(config.package.Package):
   def __init__(self,framework):
     config.package.Package.__init__(self,framework)
-    self.gitcommit              = 'f0a1542a8c408e1da02478a9843bb12815a680ae' # main sep-21-2021
+    self.gitcommit              = '5c0d1f88150bf6849593dd14e412e7d3cecabd8f' # main oct-01-2021
     self.download               = ['git://https://github.com/htool-ddm/htool','https://github.com/htool-ddm/htool/archive/'+self.gitcommit+'.tar.gz']
     self.minversion             = '0.7.0'
     self.versionname            = 'HTOOL_VERSION'
@@ -11,7 +11,7 @@ class Configure(config.package.Package):
     self.minCxxVersion          = 'c++11'
     self.cxx                    = 1
     self.functions              = []
-    self.includes               = ['htool/htool.hpp']
+    self.includes               = ['htool/misc/define.hpp'] # no C++11 in this header
     self.skippackagewithoptions = 1
     self.precisions             = ['double'] # coordinates are stored in double precision, other scalars are templated, just enforce PetscReal == double during ./configure, for now
     self.usesopenmp             = 'yes'
