diff --git a/config/BuildSystem/config/packages/SuperLU_DIST.py b/config/BuildSystem/config/packages/SuperLU_DIST.py
index 4e2b4b939d..b43343894b 100644
--- a/config/BuildSystem/config/packages/SuperLU_DIST.py
+++ b/config/BuildSystem/config/packages/SuperLU_DIST.py
@@ -51,7 +51,10 @@ class Configure(config.package.CMakePackage):
     args.append('-DTPL_BLAS_LIBRARIES="'+self.libraries.toString(self.blasLapack.dlib)+'"')
     args.append('-DTPL_LAPACK_LIBRARIES="'+self.libraries.toString(self.blasLapack.dlib)+'"')
     if self.parmetis.found:
-      args.append('-DTPL_PARMETIS_INCLUDE_DIRS="'+';'.join(self.parmetis.dinclude)+'"')
+      args.append('-DTPL_PARMETIS_INCLUDE_DIRS="/"')
+      args = self.addArgStartsWith(args,'-DCMAKE_C_FLAGS:STRING','-I'+' -I'.join(self.parmetis.dinclude))
+      args = self.addArgStartsWith(args,'-DCMAKE_C_DEBUG_FLAGS:STRING','-I'+' -I'.join(self.parmetis.dinclude))
+      args = self.addArgStartsWith(args,'-DCMAKE_C_RELEASE_FLAGS:STRING','-I'+' -I'.join(self.parmetis.dinclude))
       args.append('-DTPL_PARMETIS_LIBRARIES="'+self.libraries.toString(self.parmetis.dlib)+'"')
     else:
       args.append('-Denable_parmetislib=FALSE')
