diff --git b/config/BuildSystem/config/packages/SuiteSparse.py a/config/BuildSystem/config/packages/SuiteSparse.py
index e9c7c7d3b2..3de634af59 100644
--- b/config/BuildSystem/config/packages/SuiteSparse.py
+++ a/config/BuildSystem/config/packages/SuiteSparse.py
@@ -4,7 +4,7 @@ class Configure(config.package.Package):
   def __init__(self, framework):
     config.package.Package.__init__(self,framework)
     self.minversion        = '5.6.0'
-    self.version           = '5.8.1'
+    self.version           = '5.7.1'
     self.versioninclude    = 'SuiteSparse_config.h'
     self.versionname       = 'SUITESPARSE_MAIN_VERSION.SUITESPARSE_SUB_VERSION.SUITESPARSE_SUBSUB_VERSION'
     self.gitcommit         = 'v'+self.version
@@ -59,7 +59,7 @@ class Configure(config.package.Package):
       ldflags=self.getDynamicLinkerFlags()
     else:
       ldflags=''
-    ldflags += ' '+self.setCompilers.LDFLAGS
+    ldflags+=self.setCompilers.LDFLAGS
     # SuiteSparse 5.6.0 makefile has a bug in how it treats LDFLAGS (not using the override directive)
     ldflags+=" -L\$(INSTALL_LIB)"
     self.popLanguage()
@@ -81,9 +81,6 @@ class Configure(config.package.Package):
     args.append('INSTALL_DOC='+self.installDir+'/share/doc/suitesparse')
     args.append('BLAS="'+self.libraries.toString(self.blasLapack.dlib)+'"')
     args.append('LAPACK="'+self.libraries.toString(self.blasLapack.dlib)+'"')
-    # fix for bug in SuiteSparse
-    if self.setCompilers.isDarwin(self.log):
-      args.append('LDLIBS=""')
     if self.blasLapack.mangling == 'underscore':
       flg = ''
     elif self.blasLapack.mangling == 'caps':
@@ -129,7 +126,6 @@ class Configure(config.package.Package):
       args.append('CF="'+cflags+'"')
       args.append('CHOLMOD_CONFIG="'+flg+'"')
       args.append('CUDA=no')
-      args.append('CUDA_PATH=')
 
     args = ' '.join(args)
     conffile = os.path.join(self.packageDir,self.package+'.petscconf')
