diff --color -Naur petsc-3.23.2/config/BuildSystem/config/packages/PTScotch.py petsc-3.23.2-p1/config/BuildSystem/config/packages/PTScotch.py
--- petsc-3.23.2/config/BuildSystem/config/packages/PTScotch.py	2025-03-26 15:00:03
+++ petsc-3.23.2-p1/config/BuildSystem/config/packages/PTScotch.py	2025-03-26 17:06:45
@@ -63,4 +63,10 @@

     args.append('-DBUILD_FORTRAN:BOOL=OFF')
     args.append('-DENABLE_TESTS:BOOL=OFF')
+    mpicc = self.framework.getMakeMacro('MPICC_SHOW')
+    mpicxx = self.framework.getMakeMacro('MPICXX_SHOW')
+    if mpicc and mpicxx:
+        args = self.rmArgsStartsWith(args,['-DCMAKE_CXX_COMPILER','-DCMAKE_C_COMPILER'])
+        args.append('-DCMAKE_C_COMPILER="'+mpicc.split(None, 1)[0]+'"')
+        args.append('-DCMAKE_CXX_COMPILER="'+mpicxx.split(None, 1)[0]+'"')
     return args
diff --color -Naur petsc-3.23.2/config/BuildSystem/config/packages/SuiteSparse.py petsc-3.23.2-p1/config/BuildSystem/config/packages/SuiteSparse.py
--- petsc-3.23.2/config/BuildSystem/config/packages/SuiteSparse.py	2025-03-26 15:00:03
+++ petsc-3.23.2-p1/config/BuildSystem/config/packages/SuiteSparse.py	2025-03-26 17:07:16
@@ -70,4 +70,10 @@
     args.append('-DSUITESPARSE_USE_OPENMP:BOOL='+('ON' if self.openmp.found else 'OFF'))
     args.append('-DSUITESPARSE_USE_64BIT_BLAS:BOOL='+('ON' if self.blasLapack.has64bitindices else 'OFF'))

+    mpicc = self.framework.getMakeMacro('MPICC_SHOW')
+    mpicxx = self.framework.getMakeMacro('MPICXX_SHOW')
+    if mpicc and mpicxx:
+        args = self.rmArgsStartsWith(args,['-DCMAKE_CXX_COMPILER','-DCMAKE_C_COMPILER'])
+        args.append('-DCMAKE_C_COMPILER="'+mpicc.split(None, 1)[0]+'"')
+        args.append('-DCMAKE_CXX_COMPILER="'+mpicxx.split(None, 1)[0]+'"')
     return args
diff --color -Naur petsc-3.23.2/config/BuildSystem/config/packages/SuperLU.py petsc-3.23.2-p1/config/BuildSystem/config/packages/SuperLU.py
--- petsc-3.23.2/config/BuildSystem/config/packages/SuperLU.py	2025-03-20 05:44:35
+++ petsc-3.23.2-p1/config/BuildSystem/config/packages/SuperLU.py	2025-03-26 17:07:16
@@ -54,5 +54,11 @@
       if item.find('CMAKE_C_FLAGS') >= 0 or item.find('CMAKE_CXX_FLAGS') >= 0:
         args[place]=item[:-1]+' '+mangledef+'"'

+    mpicc = self.framework.getMakeMacro('MPICC_SHOW')
+    mpicxx = self.framework.getMakeMacro('MPICXX_SHOW')
+    if mpicc and mpicxx:
+        args = self.rmArgsStartsWith(args,['-DCMAKE_CXX_COMPILER','-DCMAKE_C_COMPILER'])
+        args.append('-DCMAKE_C_COMPILER="'+mpicc.split(None, 1)[0]+'"')
+        args.append('-DCMAKE_CXX_COMPILER="'+mpicxx.split(None, 1)[0]+'"')
     return args

diff --color -Naur petsc-3.23.2/config/BuildSystem/config/packages/metis.py petsc-3.23.2-p1/config/BuildSystem/config/packages/metis.py
--- petsc-3.23.2/config/BuildSystem/config/packages/metis.py	2025-03-08 23:51:15
+++ petsc-3.23.2-p1/config/BuildSystem/config/packages/metis.py	2025-03-26 17:07:16
@@ -44,6 +44,12 @@
     if self.framework.argDB['download-metis-use-doubleprecision']:
       args.append('-DMETIS_USE_DOUBLEPRECISION=1')
     args.append('-DMATH_LIB="'+self.libraries.toStringNoDupes(self.mathlib.lib)+'"')
+    mpicc = self.framework.getMakeMacro('MPICC_SHOW')
+    mpicxx = self.framework.getMakeMacro('MPICXX_SHOW')
+    if mpicc and mpicxx:
+        args = self.rmArgsStartsWith(args,['-DCMAKE_CXX_COMPILER','-DCMAKE_C_COMPILER'])
+        args.append('-DCMAKE_C_COMPILER="'+mpicc.split(None, 1)[0]+'"')
+        args.append('-DCMAKE_CXX_COMPILER="'+mpicxx.split(None, 1)[0]+'"')
     return args

   def configureLibrary(self):
diff --color -Naur petsc-3.23.2/config/BuildSystem/config/packages/mmg.py petsc-3.23.2-p1/config/BuildSystem/config/packages/mmg.py
--- petsc-3.23.2/config/BuildSystem/config/packages/mmg.py	2025-03-26 15:00:03
+++ petsc-3.23.2-p1/config/BuildSystem/config/packages/mmg.py	2025-03-26 17:07:16
@@ -29,6 +29,12 @@
     args.append('-DUSE_VTK=OFF')
     args.append('-DMMG_INSTALL_PRIVATE_HEADERS=ON')
     args.append('-DSCOTCH_DIR:STRING="'+self.ptscotch.directory+'"')
+    mpicc = self.framework.getMakeMacro('MPICC_SHOW')
+    mpicxx = self.framework.getMakeMacro('MPICXX_SHOW')
+    if mpicc and mpicxx:
+        args = self.rmArgsStartsWith(args,['-DCMAKE_CXX_COMPILER','-DCMAKE_C_COMPILER'])
+        args.append('-DCMAKE_C_COMPILER="'+mpicc.split(None, 1)[0]+'"')
+        args.append('-DCMAKE_CXX_COMPILER="'+mpicxx.split(None, 1)[0]+'"')
     if self.getDefaultIndexSize() == 64:
       int64_t = '''
 #if !(defined(PETSC_HAVE_STDINT_H) && defined(PETSC_HAVE_INTTYPES_H) && defined(PETSC_HAVE_MPI_INT64_T))
diff --color -Naur petsc-3.23.2/config/BuildSystem/config/package.py petsc-3.23.2-p1/config/BuildSystem/config/package.py
--- petsc-3.23.2/config/BuildSystem/config/package.py	2025-03-26 15:00:03
+++ petsc-3.23.2-p1/config/BuildSystem/config/package.py	2025-03-26 17:07:16
@@ -432,7 +432,7 @@ class Package(config.base.Configure):
     if self.requirekandr:
       outflags += self.setCompilers.KandRFlags
     with self.Language('C'):
-      if self.brokengnu23 and config.setCompilers.Configure.isGcc150plus(self.getCompiler(), self.log):
+      if self.brokengnu23 and (config.setCompilers.Configure.isGcc150plus(self.getCompiler(), self.log) or config.setCompilers.Configure.isMINGW(self.getCompiler(), self.log)):
         outflags.append('-std=gnu17')
     return ' '.join(outflags)

